diff -rupN linux.orig/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi linux/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
--- linux.orig/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi	2024-02-01 18:30:30.720485939 +0000
+++ linux/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi	2024-02-01 19:36:17.927420782 +0000
@@ -33,6 +33,10 @@
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1125000 1125000 1250000>;
 		};
+		opp06 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1300000 1300000 1300000>;
+		};
 	};
 
 	cluster1_opp: opp-table-1 {
@@ -72,6 +76,10 @@
 			opp-hz = /bits/ 64 <1800000000>;
 			opp-microvolt = <1200000 1200000 1250000>;
 		};
+		opp08 {
+			opp-hz = /bits/ 64 <2088000000>;
+			opp-microvolt = <1300000 1300000 1300000>;
+		};
 	};
 
 	gpu_opp_table: opp-table-2 {
@@ -101,6 +109,23 @@
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <1100000 1100000 1150000>;
 		};
+                opp06 {
+                        opp-hz = /bits/ 64 <900000000>;
+                        opp-microvolt = <1150000 1150000 1150000>;
+                };
+	};
+
+	dmc_opp_table: opp-table-3 {
+		compatible = "operating-points-v2";
+
+		opp00 {
+			opp-hz = /bits/ 64 <666000000>;
+			opp-microvolt = <900000>;
+		};
+		opp01 {
+			opp-hz = /bits/ 64 <856000000>;
+			opp-microvolt = <900000>;
+		};
 	};
 };
 
