diff --git a/vita3k/CMakeLists.txt b/vita3k/CMakeLists.txt
index 04177f04..4adb1e0b 100644
--- a/vita3k/CMakeLists.txt
+++ b/vita3k/CMakeLists.txt
@@ -136,10 +136,6 @@ if(USE_DISCORD_RICH_PRESENCE)
 	target_link_libraries(vita3k PRIVATE discord-rpc)
 endif()
 
-if(LINUX)
-	target_link_libraries(vita3k PRIVATE -static-libgcc -static-libstdc++)
-endif()
-
 set_target_properties(vita3k PROPERTIES OUTPUT_NAME Vita3K
 	ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
 	LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
