diff --git a/configure.ac b/configure.ac
index 9551f0c..0fde264 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,7 +7,7 @@ AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([foreign dist-xz])
-
+LT_INIT
 XCB_UTIL_M4_WITH_INCLUDE_PATH
 XCB_UTIL_COMMON([1.4], [1.6])
 
diff --git a/ewmh/Makefile.am b/ewmh/Makefile.am
index 488d006..747d042 100644
--- a/ewmh/Makefile.am
+++ b/ewmh/Makefile.am
@@ -1,4 +1,4 @@
-
+pkgconfigdir = $(libdir)/pkgconfig
 lib_LTLIBRARIES = libxcb-ewmh.la
 
 nodist_xcbinclude_HEADERS = xcb_ewmh.h
diff --git a/icccm/Makefile.am b/icccm/Makefile.am
index 8dc2680..5b56c14 100644
--- a/icccm/Makefile.am
+++ b/icccm/Makefile.am
@@ -1,4 +1,4 @@
-
+pkgconfigdir = $(libdir)/pkgconfig
 lib_LTLIBRARIES = libxcb-icccm.la
 
 xcbinclude_HEADERS = xcb_icccm.h

