diff --git a/Makefile b/Makefile
index 71aadc9..b93381a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,16 +1,8 @@
 SHELL := /bin/sh
-KVER  ?= $(shell uname -r)
-KSRC := /lib/modules/$(KVER)/build
 FIRMWAREDIR := /lib/firmware/
 PWD := $(shell pwd)
 CLR_MODULE_FILES := *.mod.c *.mod *.o .*.cmd *.ko *~ .tmp_versions* modules.order Module.symvers
 SYMBOL_FILE := Module.symvers
-# Handle the move of the entire rtw88 tree
-ifneq ("","$(wildcard /lib/modules/$(KVER)/kernel/drivers/net/wireless/realtek)")
-MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/realtek/rtw88
-else
-MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/rtw88
-endif
 
 ifneq ("$(INSTALL_MOD_PATH)", "")
 DEPMOD_ARGS = -b $(INSTALL_MOD_PATH)
