diff -rupN u-boot.orig/arch/arm/dts/rk3399-evb.dts u-boot/arch/arm/dts/rk3399-evb.dts
--- u-boot.orig/arch/arm/dts/rk3399-evb.dts	2024-01-19 17:53:16.898246406 +0000
+++ u-boot/arch/arm/dts/rk3399-evb.dts	2024-01-19 17:59:15.431212367 +0000
@@ -6,6 +6,8 @@
 /dts-v1/;
 #include <dt-bindings/pwm/pwm.h>
 #include "rk3399.dtsi"
+#include "rk3399-u-boot.dtsi"
+#include "rk3399-sdram-lpddr4-100.dtsi"
 
 / {
 	model = "Rockchip RK3399 Evaluation Board";
