diff --git a/Core/System.cpp b/Core/System.cpp
index cb6fc34fa..829a5795a 100644
--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -619,12 +619,7 @@ std::string PSP_GetLoading() {
 Path GetSysDirectory(PSPDirectories directoryType) {
 	const Path &memStickDirectory = g_Config.memStickDirectory;
 	Path pspDirectory;
-	if (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP")) {
-		// Let's strip this off, to easily allow choosing a root directory named "PSP" on Android.
-		pspDirectory = memStickDirectory;
-	} else {
-		pspDirectory = memStickDirectory / "PSP";
-	}
+	pspDirectory = memStickDirectory / "PSP";
 
 	switch (directoryType) {
 	case DIRECTORY_PSP:
