diff --git a/source/common/menu/menu.cpp b/source/common/menu/menu.cpp
index 89017d67b..8c84c0679 100644
--- a/source/common/menu/menu.cpp
+++ b/source/common/menu/menu.cpp
@@ -692,20 +692,20 @@ bool M_Responder (event_t *ev)
 			ch = ev->data1;
 			switch (ch)
 			{
-			case KEY_JOY1:
-			case KEY_JOY3:
+			case KEY_JOY2:
+			case KEY_JOY4:
 			case KEY_JOY15:
 			case KEY_PAD_A:
 				mkey = MKEY_Enter;
 				break;
 
-			case KEY_JOY2:
+			case KEY_JOY1:
 			case KEY_JOY14:
 			case KEY_PAD_B:
 				mkey = MKEY_Back;
 				break;
 
-			case KEY_JOY4:
+			case KEY_JOY3:
 			case KEY_PAD_X:
 				mkey = MKEY_Clear;
 				break;
@@ -720,6 +720,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_PageDown;
 				break;
 
+			case KEY_JOY14:
 			case KEY_PAD_DPAD_UP:
 			case KEY_PAD_LTHUMB_UP:
 			case KEY_JOYAXIS2MINUS:
@@ -727,6 +728,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Up;
 				break;
 
+			case KEY_JOY15:
 			case KEY_PAD_DPAD_DOWN:
 			case KEY_PAD_LTHUMB_DOWN:
 			case KEY_JOYAXIS2PLUS:
@@ -734,6 +736,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Down;
 				break;
 
+			case KEY_JOY16:
 			case KEY_PAD_DPAD_LEFT:
 			case KEY_PAD_LTHUMB_LEFT:
 			case KEY_JOYAXIS1MINUS:
@@ -741,6 +744,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Left;
 				break;
 
+			case KEY_JOY17:
 			case KEY_PAD_DPAD_RIGHT:
 			case KEY_PAD_LTHUMB_RIGHT:
 			case KEY_JOYAXIS1PLUS:
