diff -rupN flycast.orig/Makefile flycast/Makefile
--- flycast.orig/Makefile	2022-04-08 18:10:23.156125702 -0400
+++ flycast/Makefile	2022-04-08 18:48:44.236265589 -0400
@@ -610,6 +610,24 @@ else ifneq (,$(findstring RK,$(platform)
   WITH_DYNAREC=arm
   HAVE_GENERIC_JIT = 0
 
+else ifeq ($(platform), RG552)
+        EXT ?= so
+        TARGET := $(TARGET_NAME)_libretro.$(EXT)
+        SHARED += -shared -Wl,--version-script=link.T
+        fpic = -fPIC
+        LIBS += -lrt
+        ARM_FLOAT_ABI_HARD = 0
+        FORCE_GLES = 1
+        SINGLE_PREC_FLAGS = 1
+        FORCE_GLES = 1
+        CPUFLAGS += -DTARGET_LINUX_ARMv8 -frename-registers
+        CFLAGS += -mno-outline-atomics -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72 $(CPUFLAGS)
+        CXXFLAGS += -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72 $(CPUFLAGS)
+        ASFLAGS += -march=armv8-a+crc -mcpu=cortex-a72 -c
+        WITH_DYNAREC=arm64
+        PLATFORM_EXT := unix
+        HAVE_GENERIC_JIT = 0
+
 # RockPro64
 else ifeq ($(platform), rockpro64)
 	EXT ?= so
