diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index b22db26e..ad856d93 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -315,6 +315,9 @@ std::string FileSystem::GetDisplayNameFromPath(const std::string_view& path)
 
 std::string_view Path::GetDirectory(const std::string_view& path)
 {
+
+return "/storage/.config/duckstation";
+
   const std::string::size_type pos = GetLastSeperatorPosition(path, false);
   if (pos == std::string_view::npos)
     return {};
