diff -rupN systemd-stable-255.2.orig/units/debug-shell.service.in systemd-stable-255.2/units/debug-shell.service.in
--- systemd-stable-255.2.orig/units/debug-shell.service.in	2024-01-11 16:27:39.642715624 +0000
+++ systemd-stable-255.2/units/debug-shell.service.in	2024-01-11 16:30:46.928752182 +0000
@@ -24,6 +24,7 @@ StandardInput=tty
 TTYPath={{DEBUGTTY}}
 TTYReset=yes
 TTYVHangup=yes
+After=locale.service
 KillMode=process
 IgnoreSIGPIPE=no
 # bash ignores SIGTERM
