diff -rupN systemd-stable-255.2.orig/meson.build systemd-stable-255.2/meson.build
--- systemd-stable-255.2.orig/meson.build       2024-01-11 16:27:39.478706802 +0000
+++ systemd-stable-255.2/meson.build    2024-01-11 16:28:08.724278853 +0000
@@ -254,7 +254,7 @@ conf.set_quoted('SYSTEMD_TTY_ASK_PASSWOR
 conf.set_quoted('SYSTEMD_UPDATE_HELPER_PATH',                 libexecdir / 'systemd-update-helper')
 conf.set_quoted('SYSTEMD_USERWORK_PATH',                      libexecdir / 'systemd-userwork')
 conf.set_quoted('SYSTEMD_VERITYSETUP_PATH',                   libexecdir / 'systemd-veritysetup')
-conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     pkgsysconfdir / 'system')
+conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     '/storage/.config/system.d')
 conf.set_quoted('SYSTEM_DATA_UNIT_DIR',                       systemunitdir)
 conf.set_quoted('SYSTEM_ENV_GENERATOR_DIR',                   systemenvgeneratordir)
 conf.set_quoted('SYSTEM_GENERATOR_DIR',                       systemgeneratordir)
