diff -rupN systemd-stable-253.16.orig/units/debug-shell.service.in systemd-stable-253.16/units/debug-shell.service.in
--- systemd-stable-253.16.orig/units/debug-shell.service.in     2024-01-26 14:53:42.000000000 -0500
+++ systemd-stable-253.16/units/debug-shell.service.in  2024-02-14 12:25:50.660872467 -0500
@@ -13,6 +13,7 @@ Documentation=man:systemd-debug-generato
 DefaultDependencies=no
 IgnoreOnIsolate=yes
 ConditionPathExists={{DEBUGTTY}}
+After=locale.service

 [Service]
 Environment=TERM=linux
