diff --git a/src/input_common/drivers/sdl_driver.cpp b/src/input_common/drivers/sdl_driver.cpp
index 45ce588f0..402b4956c 100644
--- a/src/input_common/drivers/sdl_driver.cpp
+++ b/src/input_common/drivers/sdl_driver.cpp
@@ -211,8 +211,8 @@ public:
     std::string GetControllerName() const {
         if (sdl_controller) {
             switch (SDL_GameControllerGetType(sdl_controller.get())) {
-            case SDL_CONTROLLER_TYPE_XBOX360:
-                return "Xbox 360 Controller";
+           // case SDL_CONTROLLER_TYPE_XBOX360:
+             //   return "Xbox 360 Controller";
             case SDL_CONTROLLER_TYPE_XBOXONE:
                 return "Xbox One Controller";
             case SDL_CONTROLLER_TYPE_PS3:
