diff --git a/Makefile b/Makefile
index 700146a..f853845 100644
--- a/Makefile
+++ b/Makefile
@@ -87,6 +87,36 @@ else ifneq (,$(findstring rockchip,$(platform)))
        GLES31 := 1
    endif
 
+# Anbernic RG552
+else ifeq ($(platform), RG552)
+    TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+	CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math -DARM
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
+    GLES := 1
+    GLES31 := 1
+
+# Anbernic RG503, RG353p
+else ifeq ($(platform), RK3566)
+    TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 -ffast-math -DARM
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
+    GLES := 1
+    GLES31 := 1
+
+# Anbernic RG351P/M, RG351MP, RG351V, RGB20S
+else ifeq ($(platform), RK3326)
+    TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+	CFLAGS += -mcpu=cortex-a35 -mtune=cortex-a35 -ffast-math -DARM
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
+    GLES := 1
+    GLES31 := 1
+
 else ifneq (,$(findstring osx,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.dylib
    fpic := -fPIC
