diff -rupN pcsx_rearmed.orig/Makefile.libretro pcsx_rearmed/Makefile.libretro
--- pcsx_rearmed.orig/Makefile.libretro        2022-03-22 22:45:18.406511156 +0000
+++ pcsx_rearmed/Makefile.libretro     2022-03-23 00:58:28.073644000 +0000
@@ -362,6 +362,19 @@ else ifeq ($(platform), rpi4_64)
       DYNAREC = ari64
         fpic := -fPIC
         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72 -ftree-vectorize
+
+#######################################
+# Anbernic RG351x
+else ifeq ($(platform), RG351x)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CFLAGS += -marm -mtune=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+   ASFLAGS += -mtune=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+   fpic := -fPIC
+   HAVE_NEON = 1
+   ARCH = arm
+   BUILTIN_GPU = neon
+   DYNAREC = ari64
+

 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>

