diff --git a/src/duckstation-nogui/nogui_host_interface.cpp b/src/duckstation-nogui/nogui_host_interface.cpp
index a8e4ddf2..eda7baa3 100644
--- a/src/duckstation-nogui/nogui_host_interface.cpp
+++ b/src/duckstation-nogui/nogui_host_interface.cpp
@@ -105,10 +105,10 @@ bool NoGUIHostInterface::CreateDisplay(bool fullscreen)
   Assert(!m_display);
   switch (g_settings.gpu_renderer)
   {
-    case GPURenderer::HardwareVulkan:
+/*    case GPURenderer::HardwareVulkan:
       m_display = std::make_unique<FrontendCommon::VulkanHostDisplay>();
       break;
-
+*/
     case GPURenderer::HardwareOpenGL:
 #ifndef _WIN32
     default:
