diff -rupN ncurses-6.4.orig/configure ncurses-6.4/configure
--- ncurses-6.4.orig/configure	2022-11-05 21:13:26.000000000 +0000
+++ ncurses-6.4/configure	2023-05-09 11:21:38.765534941 +0000
@@ -4236,7 +4236,7 @@ echo $ECHO_N "checking for first directo
 	cf_pkg_config_path=none
 	for cf_config in $cf_search_path
 	do
-		if test -d "$cf_config"
+		if test -n "$cf_config"
 		then
 			cf_pkg_config_path=$cf_config
 			break
