diff -rupN u-boot.orig/Makefile u-boot/Makefile
--- u-boot.orig/Makefile	2023-05-11 17:08:32.613932080 +0000
+++ u-boot/Makefile	2023-05-11 17:08:01.428935587 +0000
@@ -360,6 +360,9 @@ KBUILD_CPPFLAGS := -D__KERNEL__ -D__UBOO
 KBUILD_CFLAGS   := -Wall -Wstrict-prototypes \
 		   -Wno-format-security \
 		   -fno-builtin -ffreestanding
+KBUILD_CFLAGS += $(call cc-disable-warning, address-of-packed-member)
+KBUILD_CFLAGS   += $(call cc-disable-warning, dangling-pointer)
+KBUILD_CFLAGS   += $(call cc-disable-warning, address)
 KBUILD_CFLAGS	+= -fshort-wchar -Werror
 KBUILD_AFLAGS   := -D__ASSEMBLY__
 
diff -rupN u-boot.orig/arch/arm/dts/Makefile u-boot/arch/arm/dts/Makefile
--- u-boot.orig/arch/arm/dts/Makefile	2023-05-11 17:08:32.629932591 +0000
+++ u-boot/arch/arm/dts/Makefile	2023-05-11 17:08:01.448936226 +0000
@@ -28,11 +28,9 @@ dtb-$(CONFIG_EXYNOS5) += exynos5250-arnd
 	exynos5422-odroidxu3.dtb
 dtb-$(CONFIG_EXYNOS7420) += exynos7420-espresso7420.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += \
-	rk3036-sdk.dtb \
 	rk3066a-mk808.dtb \
 	rk3126-evb.dtb \
 	rk3128-evb.dtb \
-	rk3188-radxarock.dtb \
 	rk3288-evb.dtb \
 	rk3288-fennec.dtb \
 	rk3288-firefly.dtb \
@@ -46,11 +44,6 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rk3288-veyron-minnie.dtb \
 	rk3288-vyasa.dtb \
 	rk3308-evb.dtb \
-	rk3328-evb.dtb \
-	rk3368-lion.dtb \
-	rk3368-sheep.dtb \
-	rk3368-geekbox.dtb \
-	rk3368-px5-evb.dtb \
 	rk3399-evb.dtb \
 	rk3399-firefly.dtb \
 	rk3399-puma-ddr1333.dtb \
diff -rupN u-boot.orig/board/rockchip/odroidgoa/odroidgoa.c u-boot/board/rockchip/odroidgoa/odroidgoa.c
--- u-boot.orig/board/rockchip/odroidgoa/odroidgoa.c	2023-05-11 17:08:32.685934380 +0000
+++ u-boot/board/rockchip/odroidgoa/odroidgoa.c	2023-05-11 17:08:01.500937888 +0000
@@ -21,6 +21,8 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+#define RUMBLE_GPIO	15 /* GPIO0_B7 */
+
 #define ALIVE_LED_GPIO	17 /* GPIO0_C1 */
 #define WIFI_EN_GPIO	110 /* GPIO3_B6 */
 
@@ -36,6 +38,14 @@ bool is_odroidgo3(void)
 		return false;
 }
 
+void board_rumble(void)
+{
+	gpio_request(RUMBLE_GPIO, "rumble");
+	gpio_direction_output(RUMBLE_GPIO, 0);
+	gpio_free(RUMBLE_GPIO);
+}
+
+
 void board_alive_led(void)
 {
 	gpio_request(ALIVE_LED_GPIO, "alive_led");
@@ -160,8 +170,9 @@ int rk_board_late_init(void)
 	if (is_odroidgo3())
 		disp_offs = 9;
 
+	board_rumble();
 	/* turn on blue led */
-	board_alive_led();
+	//board_alive_led();
 
 	/* set wifi_en as default high */
 	if (!is_odroidgo3())
@@ -194,10 +205,10 @@ int rk_board_late_init(void)
 #endif
 
 	/* show boot logo and version */
-	lcd_show_logo();
-	lcd_setfg_color("grey");
-	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
-	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
+	//lcd_show_logo();
+	//lcd_setfg_color("red");
+	//lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
+	//lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
 
 	if (!board_check_autotest()) {
 		board_run_autotest();
diff -rupN u-boot.orig/scripts/dtc/dtc-lexer.l u-boot/scripts/dtc/dtc-lexer.l
--- u-boot.orig/scripts/dtc/dtc-lexer.l	2023-05-11 17:08:32.809938343 +0000
+++ u-boot/scripts/dtc/dtc-lexer.l	2023-05-11 17:08:01.628941978 +0000
@@ -38,7 +38,6 @@ LINECOMMENT	"//".*\n
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
diff -rupN u-boot.orig/scripts/dtc/dtc-lexer.lex.c_shipped u-boot/scripts/dtc/dtc-lexer.lex.c_shipped
--- u-boot.orig/scripts/dtc/dtc-lexer.lex.c_shipped	2023-05-11 17:08:32.809938343 +0000
+++ u-boot/scripts/dtc/dtc-lexer.lex.c_shipped	2023-05-11 17:08:01.628941978 +0000
@@ -631,7 +631,6 @@ char *yytext;
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
