diff --git a/Makefile b/Makefile
index c60ad289a..f688dfbf3 100644
--- a/Makefile
+++ b/Makefile
@@ -605,6 +605,8 @@ ifeq ($(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-stack-usage.sh $(CC)),y)
 endif
 
 KBUILD_CFLAGS += $(call cc-option,-Wno-format-nonliteral)
+KBUILD_CFLAGS   += $(call cc-disable-warning, maybe-uninitialized)
+KBUILD_CFLAGS   += $(call cc-disable-warning, enum-int-mismatch)
 
 # turn jbsr into jsr for m68k
 ifeq ($(ARCH),m68k)
