diff -rupN uboot.orig/cmd/hwrev.c uboot/cmd/hwrev.c
--- uboot.orig/cmd/hwrev.c	2024-01-04 18:59:49.400779885 +0000
+++ uboot/cmd/hwrev.c	2024-01-04 19:02:14.336283106 +0000
@@ -34,6 +34,21 @@ int do_hwrev(cmd_tbl_t *cmdtp, int flag,
 		env_set("hwrev", "v10-go3");
 		env_set("dtb_name", "rk3326-odroidgo3-linux.dtb");
 	}
+        /* RG351V */
+        else if (check_range(490, 540, hwrev_adc)) {
+                env_set("hwrev", "rg351v");
+                env_set("dtb_name", "rk3326-anbernic-rg351v.dtb");
+        }
+        /* RG351MP */
+        else if (check_range(140, 190, hwrev_adc)) {
+                env_set("hwrev", "rg351mp");
+                env_set("dtb_name", "rk3326-anbernic-rg351mp.dtb");
+        }
+        /* XU10 */
+        else if (check_range(1000, 1050, hwrev_adc)) {
+                env_set("hwrev", "xu10");
+                env_set("dtb_name", "rk3326-magicx-xu10.dtb");
+        }
 	/* engineer samples */
 	else {
 		env_set("hwrev", "v00");
