diff -rupN pcsx_rearmed.orig/Makefile.libretro pcsx_rearmed/Makefile.libretro
--- pcsx_rearmed.orig/Makefile.libretro 2022-08-16 15:19:42.928678600 +0200
+++ pcsx_rearmed/Makefile.libretro      2022-08-17 18:05:07.746725183 +0200
@@ -385,6 +385,15 @@ else ifeq ($(platform), rpi4_64)
         fpic := -fPIC
         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72 -ftree-vectorize

+else ifeq ($(platform), RG351x)
+       TARGET := $(TARGET_NAME)_libretro.so
+       ARCH := arm64
+       BUILTIN_GPU = neon
+       HAVE_NEON = 1
+       DYNAREC = ari64
+       fpic := -fPIC
+       CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a35 -ftree-vectorize -Ofast
+
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
