diff --git a/Makefile b/Makefile
index d2c956f..2010647 100644
--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,8 @@ DEVICE ?= RK3326
 DEVICE ?= RK3399
 DEVICE ?= RK3566
 DEVICE ?= RK3566_X55
+DEVICE ?= RK3566_BSP
+DEVICE ?= RK3566_BSP_X55
 DEVICE ?= RK3588
 DEVICE ?= RK3588_ACE
 DEVICE ?= S922X
diff --git a/src/def.h b/src/def.h
index 7776cf5..f4f2831 100644
--- a/src/def.h
+++ b/src/def.h
@@ -58,6 +58,22 @@
    #define KEYBOARD_KEY_SPACING     4
    #define KEYBOARD_SYMBOL_SIZE     24
 
+// Parameters for rk3566
+#elif defined(DEVICE_RK3566_BSP)
+   #define SCREEN_WIDTH             640
+   #define SCREEN_HEIGHT            480
+   #define HARDWARE_ACCELERATION    0
+   #define FULLSCREEN               1
+   #define FONT_NAME                "NotoSans-Regular.ttf"
+   #define FONT_NAME_MONO           "NotoSansMono-Regular.ttf"
+   #define FONT_SIZE                20
+   #define LINE_HEIGHT              32
+   #define ICON_SIZE                24
+   #define MARGIN_X                 10
+   #define KEYBOARD_MARGIN          8
+   #define KEYBOARD_KEY_SPACING     4
+   #define KEYBOARD_SYMBOL_SIZE     24
+
 // Parameters for rk3566-x55
 #elif defined(DEVICE_RK3566_X55)
    #define SCREEN_WIDTH             1280
@@ -74,6 +90,22 @@
    #define KEYBOARD_KEY_SPACING     10
    #define KEYBOARD_SYMBOL_SIZE     58
 
+// Parameters for rk3566-x55
+#elif defined(DEVICE_RK3566_BSP_X55)
+   #define SCREEN_WIDTH             1280
+   #define SCREEN_HEIGHT            720
+   #define HARDWARE_ACCELERATION    0
+   #define FULLSCREEN               1
+   #define FONT_NAME                "NotoSans-Regular.ttf"
+   #define FONT_NAME_MONO           "NotoSansMono-Regular.ttf"
+   #define FONT_SIZE                48
+   #define LINE_HEIGHT              72
+   #define ICON_SIZE                58
+   #define MARGIN_X                 24
+   #define KEYBOARD_MARGIN          19
+   #define KEYBOARD_KEY_SPACING     10
+   #define KEYBOARD_SYMBOL_SIZE     58
+
 // Parameters for rk3588
 #elif defined(DEVICE_RK3588)
    #define SCREEN_WIDTH             1280
@@ -182,7 +214,7 @@
 #define VIEWER_SCROLL_SPEED      20
 
 // Button events
-#if defined(DEVICE_AMD64) || defined(DEVICE_RK3326) || defined(DEVICE_RK3399) || defined(DEVICE_RK3566) || defined(DEVICE_RK3566_X55) || defined(DEVICE_RK3588) || defined(DEVICE_RK3588_ACE) || defined(DEVICE_S922X)
+#if defined(DEVICE_AMD64) || defined(DEVICE_RK3326) || defined(DEVICE_RK3399) || defined(DEVICE_RK3566) || defined(DEVICE_RK3566_X55) || defined(DEVICE_RK3566_BSP) || defined(DEVICE_RK3566_BSP_X55) || defined(DEVICE_RK3588) || defined(DEVICE_RK3588_ACE) || defined(DEVICE_S922X)
    #define BUTTON_PRESSED_UP              event.type == SDL_KEYDOWN && event.key.repeat == 0 && (event.key.keysym.sym == SDLK_UP || event.key.keysym.sym == SDLK_w)
    #define BUTTON_PRESSED_DOWN            event.type == SDL_KEYDOWN && event.key.repeat == 0 && (event.key.keysym.sym == SDLK_DOWN || event.key.keysym.sym == SDLK_s)
    #define BUTTON_PRESSED_LEFT            event.type == SDL_KEYDOWN && event.key.repeat == 0 && event.key.keysym.sym == SDLK_LEFT
