diff --git a/gfx/common/wayland_common.c b/gfx/common/wayland_common.c
index bf02a4b869..71cb9eb328 100644
--- a/gfx/common/wayland_common.c
+++ b/gfx/common/wayland_common.c
@@ -534,10 +534,10 @@ static bool wl_draw_splash_screen(gfx_ctx_wayland_data_t *wl)
 
    shm_buffer_paint_checkerboard(buffer, wl->buffer_width,
       wl->buffer_height, 1,
-      8, 0xffbcbcbc, 0xff8e8e8e);
-   shm_buffer_paint_icon(buffer, wl->buffer_width,
-      wl->buffer_height, 1,
-      16);
+      8, 0xff000000, 0xff000000);
+   //shm_buffer_paint_icon(buffer, wl->buffer_width,
+   //   wl->buffer_height, 1,
+   //   16);
 
    wl_surface_attach(wl->surface, buffer->wl_buffer, 0, 0);
    wl_surface_set_buffer_scale(wl->surface, wl->buffer_scale);
