diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index 79d3e31605..88e0e469e0 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -1778,7 +1778,7 @@ static void frontend_unix_get_env(int *argc,
             "/usr/share/games/retroarch",
             "assets", sizeof(g_defaults.dirs[DEFAULT_DIR_ASSETS]));
    else
-      fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_ASSETS], base_path,
+      fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_ASSETS], "/tmp/",
             "assets", sizeof(g_defaults.dirs[DEFAULT_DIR_ASSETS]));
 
 #if defined(DINGUX)
@@ -1847,21 +1847,21 @@ static void frontend_unix_get_env(int *argc,
    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_REMAP],
          g_defaults.dirs[DEFAULT_DIR_MENU_CONFIG],
          "remaps", sizeof(g_defaults.dirs[DEFAULT_DIR_REMAP]));
-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_PLAYLIST], base_path,
+   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_PLAYLIST], "/tmp/",
          "playlists", sizeof(g_defaults.dirs[DEFAULT_DIR_PLAYLIST]));
    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_RECORD_CONFIG], base_path,
          "records_config", sizeof(g_defaults.dirs[DEFAULT_DIR_RECORD_CONFIG]));
    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_RECORD_OUTPUT], base_path,
          "records", sizeof(g_defaults.dirs[DEFAULT_DIR_RECORD_OUTPUT]));
-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_DATABASE], base_path,
+   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_DATABASE], "/tmp/",
          "database/rdb", sizeof(g_defaults.dirs[DEFAULT_DIR_DATABASE]));
-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_SHADER], base_path,
+   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_SHADER], "/tmp/",
          "shaders", sizeof(g_defaults.dirs[DEFAULT_DIR_SHADER]));
    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_CHEATS], base_path,
          "cheats", sizeof(g_defaults.dirs[DEFAULT_DIR_CHEATS]));
-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_OVERLAY], base_path,
+   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_OVERLAY], "/tmp/",
          "overlays", sizeof(g_defaults.dirs[DEFAULT_DIR_OVERLAY]));
-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_OSK_OVERLAY], base_path,
+   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_OSK_OVERLAY], "/tmp/",
          "overlays/keyboards", sizeof(g_defaults.dirs[DEFAULT_DIR_OSK_OVERLAY]));
    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_CORE_ASSETS], base_path,
          "downloads", sizeof(g_defaults.dirs[DEFAULT_DIR_CORE_ASSETS]));
