diff -rupN a5200.orig/Makefile a5200/Makefile
--- a5200.orig/Makefile	2022-03-20 09:32:13.018172213 -0400
+++ a5200/Makefile	2022-03-20 09:33:58.556978254 -0400
@@ -363,8 +363,17 @@ else ifeq ($(platform), rpi4)
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
 	FLAGS += -DARM 
-	FLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
-	FLAGS += -fomit-frame-pointer -ffast-math
+	FLAGS += -mtune=cortex-a72.cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+	CFLAGS += -DDEFAULT_CFG_NAME="\"/storage/.config/game/configs/atari800.cfg\""
+
+# RG552
+else ifeq ($(platform), rg552)
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+        FLAGS += -DARM
+        FLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
+        FLAGS += -fomit-frame-pointer -ffast-math
 
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
