diff -rupN linux.orig/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c linux/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
--- linux.orig/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c	2024-03-07 05:53:01.096433183 +0000
+++ linux/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c	2024-03-07 05:55:10.999624906 +0000
@@ -4857,8 +4857,8 @@ static int vop2_plane_atomic_check(struc
 
 		if (vop2_cluster_window(win) && !vpstate->afbc_en &&
 		    (win->supported_rotations & state->rotation)) {
-			DRM_ERROR("Unsupported linear rotation(%d) format at %s\n",
-				  state->rotation, win->name);
+			//DRM_ERROR("Unsupported linear rotation(%d) format at %s\n",
+			//	  state->rotation, win->name);
 			return -EINVAL;
 		}
 	}
