diff -rupN flycast.orig/Makefile flycast/Makefile
--- flycast.orig/Makefile	2022-04-08 18:10:23.156125702 -0400
+++ flycast/Makefile	2022-04-08 18:59:10.972092728 -0400
@@ -610,6 +610,30 @@ else ifneq (,$(findstring RK,$(platform)
   WITH_DYNAREC=arm
   HAVE_GENERIC_JIT = 0
 
+else ifeq ($(platform), RG351x)
+        EXT ?= so
+        TARGET := $(TARGET_NAME)_libretro.$(EXT)
+        SHARED += -shared -Wl,--version-script=link.T
+        fpic = -fPIC
+        LIBS += -lrt
+        ARM_FLOAT_ABI_HARD = 0
+        FORCE_GLES = 1
+        SINGLE_PREC_FLAGS = 1
+        HAVE_LTCG = 0
+        HAVE_OPENMP = 0
+        CFLAGS += -Ofast \
+        -fuse-linker-plugin \
+        -fno-stack-protector -fno-ident -fomit-frame-pointer \
+        -fmerge-all-constants -ffast-math -funroll-all-loops \
+        -mcpu=cortex-a35 -mtune=cortex-a35
+        CXXFLAGS += $(CFLAGS)
+        LDFLAGS += -mcpu=cortex-a35 -mtune=cortex-a35 -Ofast -flto -fuse-linker-plugin
+        PLATFORM_EXT := unix
+        CORE_DEFINES += -DLOW_END -DLOW_RES
+        WITH_DYNAREC=arm64
+        PLATFORM_EXT := unix
+        HAVE_GENERIC_JIT = 0
+
 # RockPro64
 else ifeq ($(platform), rockpro64)
 	EXT ?= so
