diff --git a/plugins/sixaxis.c b/plugins/sixaxis.c
index 9a0aa5d4f..810e33353 100644
--- a/plugins/sixaxis.c
+++ b/plugins/sixaxis.c
@@ -356,6 +356,7 @@ static bool setup_device(int fd, const char *sysfs_path,
 	else
 		btd_device_set_pnpid(device, cp->source, cp->vid, cp->pid, cp->version);
 
+	btd_device_set_trusted(device, true);
 	btd_device_set_temporary(device, true);
 
 	closure = g_new0(struct authentication_closure, 1);
