diff -rupN SDL2.orig/src/video/kmsdrm/SDL_kmsdrmmouse.c SDL2/src/video/kmsdrm/SDL_kmsdrmmouse.c
--- SDL2.orig/src/video/kmsdrm/SDL_kmsdrmmouse.c	2023-07-01 17:04:05.000000000 +0000
+++ SDL2/src/video/kmsdrm/SDL_kmsdrmmouse.c	2023-07-26 10:27:12.232726116 +0000
@@ -150,6 +150,7 @@ static int KMSDRM_DumpCursorToBO(SDL_Vid
     uint8_t *src_row;
 
     int i;
+    int j;
     int ret;
 
     if (curdata == NULL || !dispdata->cursor_bo) {
@@ -170,8 +171,10 @@ static int KMSDRM_DumpCursorToBO(SDL_Vid
 
     /* Copy from the cursor buffer to a buffer that we can dump to the GBM BO. */
     for (i = 0; i < curdata->h; i++) {
-        src_row = &((uint8_t *)curdata->buffer)[i * curdata->w * 4];
-        SDL_memcpy(ready_buffer + (i * bo_stride), src_row, (size_t)4 * curdata->w);
+        for (j = 0; j < curdata->w; j++) {
+            src_row = ((uint32_t*)curdata->buffer)[i * curdata->w + j];
+            SDL_memcpy(ready_buffer + ((curdata->w - j + 1) * bo_stride) + i, &src_row, 4);
+        }
     }
 
     /* Dump the cursor buffer to our GBM BO. */
@@ -370,6 +373,7 @@ static int KMSDRM_WarpMouseGlobal(int x,
 
         SDL_Window *window = mouse->focus;
         SDL_DisplayData *dispdata = (SDL_DisplayData *)SDL_GetDisplayForWindow(window)->driverdata;
+        KMSDRM_CursorData *curdata = mouse->cur_cursor->driverdata;
 
         /* Update internal mouse position. */
         SDL_SendMouseMotion(mouse->focus, mouse->mouseID, 0, x, y);
@@ -378,7 +382,7 @@ static int KMSDRM_WarpMouseGlobal(int x,
         if (dispdata->cursor_bo) {
             int ret = 0;
 
-            ret = KMSDRM_drmModeMoveCursor(dispdata->cursor_bo_drm_fd, dispdata->crtc->crtc_id, x, y);
+            ret = KMSDRM_drmModeMoveCursor(dispdata->cursor_bo_drm_fd, dispdata->crtc->crtc_id, y, dispdata->mode.vdisplay + curdata->w - x);
 
             if (ret) {
                 SDL_SetError("drmModeMoveCursor() failed.");
@@ -432,13 +436,14 @@ static void KMSDRM_MoveCursor(SDL_Cursor
 
         SDL_Window *window = mouse->focus;
         SDL_DisplayData *dispdata = (SDL_DisplayData *)SDL_GetDisplayForWindow(window)->driverdata;
+        KMSDRM_CursorData *curdata = mouse->cur_cursor->driverdata;
 
         if (!dispdata->cursor_bo) {
             SDL_SetError("Cursor not initialized properly.");
             return;
         }
 
-        ret = KMSDRM_drmModeMoveCursor(dispdata->cursor_bo_drm_fd, dispdata->crtc->crtc_id, mouse->x, mouse->y);
+        ret = KMSDRM_drmModeMoveCursor(dispdata->cursor_bo_drm_fd, dispdata->crtc->crtc_id, mouse->y, dispdata->mode.vdisplay - curdata->w - mouse->x);
 
         if (ret) {
             SDL_SetError("drmModeMoveCursor() failed.");
Binary files SDL2.orig/src/video/kmsdrm/.SDL_kmsdrmmouse.c.rej.swp and SDL2/src/video/kmsdrm/.SDL_kmsdrmmouse.c.rej.swp differ
