diff -rupN jslisten.orig/src/jslisten.c jslisten/src/jslisten.c
--- jslisten.orig/src/jslisten.c	2022-03-31 19:24:44.150929212 -0400
+++ jslisten/src/jslisten.c	2022-03-31 19:25:43.158549592 -0400
@@ -53,8 +53,8 @@
 
 #define NAME_LENGTH 128
 #define MYPROGNAME "jslisten"
-#define myConfFile "/.jslisten"
-#define myGlConfFile "/etc/jslisten.cfg"
+#define myConfFile "/.config/system/configs/jslisten.cfg"
+#define myGlConfFile "/storage/.config/system/configs/jslisten.cfg"
 //#define MY_LOG_LEVEL LOG_NOTICE //LOG_DEBUG //LOG_NOTICE
 
 #define INI_BUFFERSIZE      512
