diff -rupN libtracefs-1.6.4.orig/include/tracefs.h libtracefs-1.6.4/include/tracefs.h
--- libtracefs-1.6.4.orig/include/tracefs.h	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/include/tracefs.h	2023-06-13 12:09:43.902967775 +0000
@@ -8,7 +8,7 @@
 
 #include <fcntl.h>
 #include <sched.h>
-#include <event-parse.h>
+#include <traceevent/event-parse.h>
 
 char *tracefs_get_tracing_file(const char *name);
 void tracefs_put_tracing_file(char *name);
diff -rupN libtracefs-1.6.4.orig/src/tracefs-events.c libtracefs-1.6.4/src/tracefs-events.c
--- libtracefs-1.6.4.orig/src/tracefs-events.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/src/tracefs-events.c	2023-06-13 12:10:21.580262836 +0000
@@ -15,7 +15,7 @@
 #include <fcntl.h>
 #include <limits.h>
 
-#include <kbuffer.h>
+#include <traceevent/kbuffer.h>
 
 #include "tracefs.h"
 #include "tracefs-local.h"
diff -rupN libtracefs-1.6.4.orig/src/tracefs-filter.c libtracefs-1.6.4/src/tracefs-filter.c
--- libtracefs-1.6.4.orig/src/tracefs-filter.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/src/tracefs-filter.c	2023-06-13 12:11:35.919870363 +0000
@@ -6,7 +6,7 @@
  * Copyright (C) 2021, VMware, Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
  *
  */
-#include <trace-seq.h>
+#include <traceevent/trace-seq.h>
 #include <stdlib.h>
 #include <ctype.h>
 #include <errno.h>
diff -rupN libtracefs-1.6.4.orig/src/tracefs-record.c libtracefs-1.6.4/src/tracefs-record.c
--- libtracefs-1.6.4.orig/src/tracefs-record.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/src/tracefs-record.c	2023-06-13 12:10:28.989358796 +0000
@@ -13,7 +13,7 @@
 #include <sys/ioctl.h>
 #include <sys/select.h>
 
-#include <kbuffer.h>
+#include <traceevent/kbuffer.h>
 
 #include "tracefs.h"
 #include "tracefs-local.h"
diff -rupN libtracefs-1.6.4.orig/src/tracefs-sqlhist.c libtracefs-1.6.4/src/tracefs-sqlhist.c
--- libtracefs-1.6.4.orig/src/tracefs-sqlhist.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/src/tracefs-sqlhist.c	2023-06-13 12:11:26.398318407 +0000
@@ -6,7 +6,7 @@
  * Copyright (C) 2021, VMware, Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
  *
  */
-#include <trace-seq.h>
+#include <traceevent/trace-seq.h>
 #include <stdlib.h>
 #include <stdarg.h>
 #include <ctype.h>
diff -rupN libtracefs-1.6.4.orig/src/tracefs-utils.c libtracefs-1.6.4/src/tracefs-utils.c
--- libtracefs-1.6.4.orig/src/tracefs-utils.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/src/tracefs-utils.c	2023-06-13 12:09:51.235958151 +0000
@@ -15,8 +15,8 @@
 #include <unistd.h>
 #include <errno.h>
 
-#include <event-parse.h>
-#include <event-utils.h>
+#include <traceevent/event-parse.h>
+#include <traceevent/event-utils.h>
 #include "tracefs.h"
 #include "tracefs-local.h"
 
diff -rupN libtracefs-1.6.4.orig/utest/tracefs-utest.c libtracefs-1.6.4/utest/tracefs-utest.c
--- libtracefs-1.6.4.orig/utest/tracefs-utest.c	2023-06-13 12:06:07.781799063 +0000
+++ libtracefs-1.6.4/utest/tracefs-utest.c	2023-06-13 12:10:13.095028045 +0000
@@ -12,7 +12,7 @@
 #include <dirent.h>
 #include <ftw.h>
 #include <libgen.h>
-#include <kbuffer.h>
+#include <traceevent/kbuffer.h>
 #include <pthread.h>
 
 #include <sys/mount.h>
