diff -rupN libtraceevent-1.7.3.orig/meson.build libtraceevent-1.7.3/meson.build
--- libtraceevent-1.7.3.orig/meson.build	2023-08-21 13:24:13.191233221 +0000
+++ libtraceevent-1.7.3/meson.build	2023-08-21 13:25:24.571128554 +0000
@@ -44,11 +44,3 @@ subdir('plugins')
 if cunit_dep.found()
     subdir('utest')
 endif
-subdir('samples')
-subdir('Documentation')
-
-custom_target(
-    'docs',
-    output: 'docs',
-    depends: [html, man],
-    command: ['echo'])
