diff -rupN mupen64plus.orig/Makefile mupen64plus/Makefile
--- mupen64plus.orig/Makefile	2022-02-27 09:19:06.080700288 -0500
+++ mupen64plus/Makefile	2022-02-27 13:27:51.390268523 -0500
@@ -238,6 +238,35 @@ else ifneq (,$(findstring osx,$(platform
 
    COREFLAGS += -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
+
+# Rockchip RK3566
+else ifneq (,$(findstring RK,$(platform)))
+   TARGET := $(TARGET_NAME)_libretro.so
+   LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
+
+   ifneq (,$(findstring RK33,$(platform)))
+      ifneq (,$(findstring RK3399,$(platform)))
+         CPUFLAGS += -march=armv8-a -mtune=cortex-a72.cortex-a53
+      else ifneq (,$(findstring RK3566,$(platform)))
+         CPUFLAGS += -march=armv8-a -mtune=cortex-a55
+         GLES = 1
+      endif
+   else ifneq (,$(findstring RK3288,$(platform)))
+      CPUFLAGS += -march=armv7ve -mtune=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
+      GLES3 = 1
+   endif
+
+   ifneq (,$(findstring mesa,$(platform)))
+      COREFLAGS += -DEGL_NO_X11
+   endif
+
+   GLES = 1
+   GL_LIB := -lGLESv2
+   HAVE_NEON = 0
+   WITH_DYNAREC := aarch64
+   COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
+   ASFLAGS = -f elf -d ELF_TYPE
+
 # iOS
 else ifneq (,$(findstring ios,$(platform)))
    ifeq ($(IOSSDK),)
