diff --git a/gptokeyb.cpp b/gptokeyb.cpp
old mode 100644
new mode 100755
index aa4c0b0..0d62f28
--- a/gptokeyb.cpp
+++ b/gptokeyb.cpp
@@ -42,8 +42,8 @@
 #include <linux/input.h>
 #include <linux/uinput.h>
 
-#include <libevdev-1.0/libevdev/libevdev-uinput.h>
-#include <libevdev-1.0/libevdev/libevdev.h>
+#include <libevdev/libevdev-uinput.h>
+#include <libevdev/libevdev.h>
 
 #include <fcntl.h>
 #include <sstream>
@@ -2284,7 +2284,7 @@ int main(int argc, char* argv[])
   const char* config_file = nullptr;
 
   config_mode = true;
-  config_file = "/emuelec/configs/gptokeyb/default.gptk";
+  config_file = "/usr/config/gptokeyb/default.gptk";
 
   // Add hotkey environment variable if available
   if (char* env_hotkey = SDL_getenv("HOTKEY")) {
