diff --git a/src/frontend-common/fullscreen_ui.cpp b/src/frontend-common/fullscreen_ui.cpp
index ce53e69f..fc0b24d5 100644
--- a/src/frontend-common/fullscreen_ui.cpp
+++ b/src/frontend-common/fullscreen_ui.cpp
@@ -244,6 +244,7 @@ static void DoChangeDiscFromFile();
 static void DoChangeDisc();
 static void DoRequestExit();
 static void DoToggleFullscreen();
+static void EnableDoToggleFullscreen();
 static void DoCheatsMenu();
 static void DoToggleAnalogMode();
 
@@ -1038,6 +1039,11 @@ void FullscreenUI::DoToggleFullscreen()
   Host::RunOnCPUThread([]() { Host::SetFullscreen(!Host::IsFullscreen()); });
 }
 
+void FullscreenUI::EnableDoToggleFullscreen()
+{
+  Host::SetFullscreen(true);
+}
+
 //////////////////////////////////////////////////////////////////////////
 // Landing Window
 //////////////////////////////////////////////////////////////////////////
@@ -1050,6 +1056,8 @@ void FullscreenUI::SwitchToLanding()
 
 void FullscreenUI::DrawLandingWindow()
 {
+  EnableDoToggleFullscreen();
+
   BeginFullscreenColumns(nullptr, 0.0f, true);
 
   if (BeginFullscreenColumnWindow(0.0f, -710.0f, "logo", UIPrimaryDarkColor))
