diff -rupN vlc.orig/modules/Makefile.in vlc/modules/Makefile.in
--- vlc.orig/modules/Makefile.in	2022-07-01 13:11:27.935958918 -0400
+++ vlc/modules/Makefile.in	2022-07-01 13:16:39.625023811 -0400
@@ -481,7 +481,6 @@ TESTS = hpack_test$(EXEEXT) hpackenc_tes
 @HAVE_WIN32_FALSE@am__append_247 = $(X_LIBS) $(X_PRE_LIBS) -lX11
 @HAVE_DARWIN_FALSE@@HAVE_WIN32_FALSE@am__append_248 = $(X_LIBS) $(X_PRE_LIBS) -lX11
 @HAVE_EVAS_TRUE@am__append_249 = libevas_plugin.la
-@HAVE_GL_TRUE@am__append_250 = libglspectrum_plugin.la
 @ENABLE_SOUT_TRUE@@HAVE_GCRYPT_TRUE@am__append_251 = libaccess_output_livehttp_plugin.la
 @ENABLE_SOUT_TRUE@am__append_252 = libaccess_output_shout_plugin.la \
 @ENABLE_SOUT_TRUE@	libaccess_output_srt_plugin.la \
@@ -2028,13 +2027,6 @@ libgles2_plugin_la_LINK = $(LIBTOOL) $(A
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(libgles2_plugin_la_CFLAGS) $(CFLAGS) \
 	$(libgles2_plugin_la_LDFLAGS) $(LDFLAGS) -o $@
-libglspectrum_plugin_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1)
-am_libglspectrum_plugin_la_OBJECTS = visualization/glspectrum.lo \
-	visualization/visual/fft.lo visualization/visual/window.lo
-libglspectrum_plugin_la_OBJECTS =  \
-	$(am_libglspectrum_plugin_la_OBJECTS)
-@HAVE_GL_TRUE@am_libglspectrum_plugin_la_rpath = -rpath $(visudir)
 libglwin32_plugin_la_DEPENDENCIES = libchroma_copy.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_5)
 am__objects_23 =  \
@@ -6507,7 +6499,6 @@ am__depfiles_remade =  \
 	video_splitter/$(DEPDIR)/clone.Plo \
 	video_splitter/$(DEPDIR)/libpanoramix_plugin_la-panoramix.Plo \
 	video_splitter/$(DEPDIR)/wall.Plo \
-	visualization/$(DEPDIR)/glspectrum.Plo \
 	visualization/$(DEPDIR)/libgoom_plugin_la-goom.Plo \
 	visualization/$(DEPDIR)/libprojectm_plugin_la-projectm.Plo \
 	visualization/$(DEPDIR)/libvsxu_plugin_la-vsxu.Plo \
@@ -6731,7 +6722,6 @@ SOURCES = $(liba52_plugin_la_SOURCES) $(
 	$(libglconv_vaapi_x11_plugin_la_SOURCES) \
 	$(libglconv_vdpau_plugin_la_SOURCES) \
 	$(libgles2_plugin_la_SOURCES) \
-	$(libglspectrum_plugin_la_SOURCES) \
 	$(libglwin32_plugin_la_SOURCES) $(libglx_plugin_la_SOURCES) \
 	$(libgme_plugin_la_SOURCES) $(libgnutls_plugin_la_SOURCES) \
 	$(libgoom_plugin_la_SOURCES) $(libgradfun_plugin_la_SOURCES) \
@@ -7130,7 +7120,6 @@ DIST_SOURCES = $(liba52_plugin_la_SOURCE
 	$(libglconv_vaapi_x11_plugin_la_SOURCES) \
 	$(libglconv_vdpau_plugin_la_SOURCES) \
 	$(libgles2_plugin_la_SOURCES) \
-	$(libglspectrum_plugin_la_SOURCES) \
 	$(libglwin32_plugin_la_SOURCES) $(libglx_plugin_la_SOURCES) \
 	$(libgme_plugin_la_SOURCES) $(libgnutls_plugin_la_SOURCES) \
 	$(libgoom_plugin_la_SOURCES) $(libgradfun_plugin_la_SOURCES) \
@@ -12696,13 +12685,7 @@ libevent_thread_la_LDFLAGS = -static
 visudir = $(pluginsdir)/visualization
 visu_LTLIBRARIES = $(am__append_250) $(LTLIBgoom) $(LTLIBprojectm) \
 	libvisual_plugin.la $(LTLIBvsxu)
-libglspectrum_plugin_la_SOURCES = \
-	visualization/glspectrum.c \
-	visualization/visual/fft.c visualization/visual/fft.h \
-	visualization/visual/window.c visualization/visual/window.h \
-	visualization/visual/window_presets.h
 
-libglspectrum_plugin_la_LIBADD = $(GL_LIBS) $(LIBM)
 libgoom_plugin_la_SOURCES = visualization/goom.c
 libgoom_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(GOOM_CFLAGS)
 libgoom_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(visudir)'
@@ -15715,8 +15698,6 @@ visualization/$(am__dirstamp):
 visualization/$(DEPDIR)/$(am__dirstamp):
 	@$(MKDIR_P) visualization/$(DEPDIR)
 	@: > visualization/$(DEPDIR)/$(am__dirstamp)
-visualization/glspectrum.lo: visualization/$(am__dirstamp) \
-	visualization/$(DEPDIR)/$(am__dirstamp)
 visualization/visual/$(am__dirstamp):
 	@$(MKDIR_P) visualization/visual
 	@: > visualization/visual/$(am__dirstamp)
@@ -15728,8 +15709,6 @@ visualization/visual/fft.lo: visualizati
 visualization/visual/window.lo: visualization/visual/$(am__dirstamp) \
 	visualization/visual/$(DEPDIR)/$(am__dirstamp)
 
-libglspectrum_plugin.la: $(libglspectrum_plugin_la_OBJECTS) $(libglspectrum_plugin_la_DEPENDENCIES) $(EXTRA_libglspectrum_plugin_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(LINK) $(am_libglspectrum_plugin_la_rpath) $(libglspectrum_plugin_la_OBJECTS) $(libglspectrum_plugin_la_LIBADD) $(LIBS)
 video_output/opengl/libglwin32_plugin_la-vout_helper.lo:  \
 	video_output/opengl/$(am__dirstamp) \
 	video_output/opengl/$(DEPDIR)/$(am__dirstamp)
@@ -21420,7 +21399,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@video_splitter/$(DEPDIR)/clone.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@video_splitter/$(DEPDIR)/libpanoramix_plugin_la-panoramix.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@video_splitter/$(DEPDIR)/wall.Plo@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@visualization/$(DEPDIR)/glspectrum.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@visualization/$(DEPDIR)/libgoom_plugin_la-goom.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@visualization/$(DEPDIR)/libprojectm_plugin_la-projectm.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@visualization/$(DEPDIR)/libvsxu_plugin_la-vsxu.Plo@am__quote@ # am--include-marker
@@ -30324,7 +30302,6 @@ distclean: distclean-recursive
 	-rm -f video_splitter/$(DEPDIR)/clone.Plo
 	-rm -f video_splitter/$(DEPDIR)/libpanoramix_plugin_la-panoramix.Plo
 	-rm -f video_splitter/$(DEPDIR)/wall.Plo
-	-rm -f visualization/$(DEPDIR)/glspectrum.Plo
 	-rm -f visualization/$(DEPDIR)/libgoom_plugin_la-goom.Plo
 	-rm -f visualization/$(DEPDIR)/libprojectm_plugin_la-projectm.Plo
 	-rm -f visualization/$(DEPDIR)/libvsxu_plugin_la-vsxu.Plo
@@ -31722,7 +31699,6 @@ maintainer-clean: maintainer-clean-recur
 	-rm -f video_splitter/$(DEPDIR)/clone.Plo
 	-rm -f video_splitter/$(DEPDIR)/libpanoramix_plugin_la-panoramix.Plo
 	-rm -f video_splitter/$(DEPDIR)/wall.Plo
-	-rm -f visualization/$(DEPDIR)/glspectrum.Plo
 	-rm -f visualization/$(DEPDIR)/libgoom_plugin_la-goom.Plo
 	-rm -f visualization/$(DEPDIR)/libprojectm_plugin_la-projectm.Plo
 	-rm -f visualization/$(DEPDIR)/libvsxu_plugin_la-vsxu.Plo
