diff --git a/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c b/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c
index 2445663d4..194605ff0 100644
--- a/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c
+++ b/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c
@@ -33,7 +33,7 @@
 
 #define GPIO_WRITEMASK(bits)	((bits) << 16)
 
-#define DTB_DIR			"rockchip/"
+#define DTB_DIR			""
 
 struct rg3xx_model {
 	const u16 adc_value;
@@ -375,7 +375,7 @@ int rgxx3_detect_device(void)
 			if (ret) {
 				if (board_id == RG353V)
 					board_id = RG353VS;
-				if (board_id == RG353P)
+				else if (board_id == RG353P)
 					board_id = RG353PS;
 				else
 					board_id = RGARCS;
diff --git a/configs/anbernic-rgxx3-rk3566_defconfig b/configs/anbernic-rgxx3-rk3566_defconfig
index 4e72f7581..27911cc9f 100644
--- a/configs/anbernic-rgxx3-rk3566_defconfig
+++ b/configs/anbernic-rgxx3-rk3566_defconfig
@@ -30,7 +30,7 @@ CONFIG_SPL_LOAD_FIT=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_STDOUT_VIA_ALIAS=y
-CONFIG_DEFAULT_FDT_FILE="rockchip/rk3566-anbernic-rgxx3.dtb"
+CONFIG_DEFAULT_FDT_FILE="rk3566-anbernic-rgxx3.dtb"
 # CONFIG_CONSOLE_MUX is not set
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_DISPLAY_BOARDINFO_LATE=y
