diff --git a/Core/Config.cpp b/Core/Config.cpp
index 008ef0c127..470ffd4d15 100644
--- a/Core/Config.cpp
+++ b/Core/Config.cpp
@@ -1392,10 +1392,10 @@ void Config::PostLoadCleanup(bool gameSpecific) {
 		g_Config.bSkipBufferEffects = false;
 	}
 
-	// Automatically silence secondary instances. Could be an option I guess, but meh.
-	if (PPSSPP_ID > 1) {
-		g_Config.iGlobalVolume = 0;
-	}
+//	// Automatically silence secondary instances. Could be an option I guess, but meh.
+//	if (PPSSPP_ID > 1) {
+//		g_Config.iGlobalVolume = 0;
+//	}
 
 	// Automatically switch away from deprecated setting value.
 	if (iTexScalingLevel <= 0) {
