diff -rupN weston.orig/clients/image.c weston/clients/image.c
--- weston.orig/clients/image.c	2024-02-05 14:42:51.052285175 +0000
+++ weston/clients/image.c	2024-02-05 14:44:09.510805841 +0000
@@ -402,7 +402,7 @@ image_create(struct display *display, co
 	widget_set_button_handler(image->widget, button_handler);
 	widget_set_axis_handler(image->widget, axis_handler);
 	window_set_key_handler(image->window, key_handler);
-	widget_schedule_resize(image->widget, 500, 400);
+	widget_schedule_resize(image->widget, 380, 320);
 
 	return image;
 }
diff -rupN weston.orig/clients/meson.build weston/clients/meson.build
--- weston.orig/clients/meson.build	2024-02-05 14:42:51.052285175 +0000
+++ weston/clients/meson.build	2024-02-05 14:47:27.792730710 +0000
@@ -270,83 +270,7 @@ foreach t : tools_list
 endforeach
 
 demo_clients = [
-	{ 'basename': 'clickdot' },
-	{
-		'basename': 'cliptest',
-		'dep_objs': [ dep_vertex_clipping, dep_matrix_c ]
-	},
-	{
-		'basename': 'constraints',
-		'add_sources': [
-			xdg_shell_client_protocol_h,
-			xdg_shell_protocol_c,
-		],
-	},
-	{
-		'basename': 'content_protection',
-		'add_sources': [
-			weston_content_protection_client_protocol_h,
-			weston_content_protection_protocol_c,
-		]
-	},
-
-	{ 'basename': 'dnd' },
-	{
-		'basename': 'editor',
-		'add_sources': [
-			text_input_unstable_v1_client_protocol_h,
-			text_input_unstable_v1_protocol_c,
-		],
-		'deps': [ 'pangocairo', 'gobject-2.0' ]
-	},
-	{ 'basename': 'eventdemo' },
-	{ 'basename': 'flower' },
-	{
-		'basename': 'fullscreen',
-		'add_sources': [
-			fullscreen_shell_unstable_v1_client_protocol_h,
-			fullscreen_shell_unstable_v1_protocol_c,
-		]
-	},
 	{ 'basename': 'image' },
-	{
-		'basename': 'multi-resource',
-		'add_sources': [
-			xdg_shell_client_protocol_h,
-			xdg_shell_protocol_c,
-		]
-	},
-	{
-		'basename': 'presentation-shm',
-		'add_sources': [
-			presentation_time_client_protocol_h,
-			presentation_time_protocol_c,
-			xdg_shell_client_protocol_h,
-			xdg_shell_protocol_c,
-		]
-	},
-	{ 'basename': 'resizor' },
-	{
-		'basename': 'scaler',
-		'add_sources': [
-			viewporter_client_protocol_h,
-			viewporter_protocol_c,
-		]
-	},
-	{ 'basename': 'smoke' },
-	{ 'basename': 'stacking' },
-	{
-		'basename': 'subsurfaces',
-		'deps': [ 'egl', 'glesv2', 'wayland-egl' ]
-	},
-	{
-		'basename': 'tablet',
-		'add_sources': [
-			tablet_unstable_v2_client_protocol_h,
-			tablet_unstable_v2_protocol_c,
-		],
-	},
-	{ 'basename': 'transformed' },
 ]
 
 if get_option('demo-clients')
