diff -rupN weston-12.0.3.orig/libweston/compositor.c weston-12.0.3/libweston/compositor.c
--- weston-12.0.3.orig/libweston/compositor.c	2024-01-15 15:42:23.723505677 +0000
+++ weston-12.0.3/libweston/compositor.c	2024-01-15 16:06:34.954783756 +0000
@@ -3349,10 +3349,10 @@ weston_output_finish_frame(struct weston
 	 * If timestamp of latest vblank is given, it must always go forwards.
 	 * If not given, INVALID flag must be set.
 	 */
-	if (stamp)
-		assert(timespec_sub_to_nsec(stamp, &output->frame_time) >= 0);
-	else
-		assert(presented_flags & WP_PRESENTATION_FEEDBACK_INVALID);
+	//if (stamp)
+	//	assert(timespec_sub_to_nsec(stamp, &output->frame_time) >= 0);
+	//else
+	//	assert(presented_flags & WP_PRESENTATION_FEEDBACK_INVALID);
 
 	weston_compositor_read_presentation_clock(compositor, &now);
 
