diff -rupN fileman.orig/src/def.h fileman/src/def.h
--- fileman.orig/src/def.h	2023-05-29 07:19:38.182586845 +0900
+++ fileman/src/def.h	2023-05-29 16:26:16.490203727 +0900
@@ -28,18 +28,18 @@
 
 // Parameters for rk3566
 #elif defined(DEVICE_RK3566)
-   #define SCREEN_WIDTH             640
-   #define SCREEN_HEIGHT            480
+   #define SCREEN_WIDTH             960
+   #define SCREEN_HEIGHT            544
    #define HARDWARE_ACCELERATION    0
    #define FULLSCREEN               1
    #define FONT_NAME                "NotoSans-Regular.ttf"
    #define FONT_NAME_MONO           "NotoSansMono-Regular.ttf"
-   #define FONT_SIZE                20
-   #define LINE_HEIGHT              32
+   #define FONT_SIZE                14
+   #define LINE_HEIGHT              24
    #define ICON_SIZE                24
-   #define MARGIN_X                 10
-   #define KEYBOARD_MARGIN          8
-   #define KEYBOARD_KEY_SPACING     4
+   #define MARGIN_X                 8
+   #define KEYBOARD_MARGIN          3
+   #define KEYBOARD_KEY_SPACING     3
    #define KEYBOARD_SYMBOL_SIZE     24
 
 // Parameters for rk3588
