diff -rupN fileman.orig/src/def.h fileman/src/def.h
--- fileman.orig/src/def.h	2023-05-23 02:29:52.438910332 +0900
+++ fileman/src/def.h	2023-05-23 02:35:50.129227751 +0900
@@ -28,8 +28,8 @@
 
 // Parameters for rk3566
 #elif defined(DEVICE_RK3566)
-   #define SCREEN_WIDTH             640
-   #define SCREEN_HEIGHT            480
+   #define SCREEN_WIDTH             960
+   #define SCREEN_HEIGHT            544
    #define HARDWARE_ACCELERATION    0
    #define FULLSCREEN               1
    #define FONT_NAME                "NotoSans-Regular.ttf"
