diff --git a/Makefile b/Makefile
index 590fd4c1f..3165ec95f 100644
--- a/Makefile
+++ b/Makefile
@@ -606,6 +606,8 @@ endif
 
 KBUILD_CFLAGS += $(call cc-option,-Wno-format-nonliteral)
 KBUILD_CFLAGS += $(call cc-disable-warning, address-of-packed-member)
+KBUILD_CFLAGS   += $(call cc-disable-warning, dangling-pointer)
+KBUILD_CFLAGS   += $(call cc-disable-warning, address)
 
 # turn jbsr into jsr for m68k
 ifeq ($(ARCH),m68k)
