diff -rupN u-boot.orig/include/configs/rk3588_common.h u-boot/include/configs/rk3588_common.h
--- u-boot.orig/include/configs/rk3588_common.h	2023-01-07 13:50:19.843002039 +0000
+++ u-boot/include/configs/rk3588_common.h	2023-01-07 13:53:45.363076196 +0000
@@ -74,7 +74,7 @@
 	BOOTENV_SHARED_MTD	\
 	ENV_MEM_LAYOUT_SETTINGS \
 	"fdtfile=" FDTFILE \
-	"partitions=" PARTS_RKIMG \
+        "partitions=" PARTS_DEFAULT \
 	ROCKCHIP_DEVICE_SETTINGS \
 	RKIMG_DET_BOOTDEV \
 	BOOTENV
diff -rupN u-boot.orig/include/configs/rockchip-common.h u-boot/include/configs/rockchip-common.h
--- u-boot.orig/include/configs/rockchip-common.h	2023-01-07 13:50:19.843002039 +0000
+++ u-boot/include/configs/rockchip-common.h	2023-01-07 13:55:21.026665180 +0000
@@ -140,7 +140,7 @@
 	"name=loader2,start=8MB,size=4MB,uuid=${uuid_gpt_loader2};" \
 	"name=trust,size=4M,uuid=${uuid_gpt_atf};" \
 	"name=boot,size=112M,bootable,uuid=${uuid_gpt_boot};" \
-	"name=rootfs,size=-,uuid="ROOT_UUID
+	"name=JELOS,start=16MB,size=3072M,bootable,uuid=${uuid_gpt_boot};\0"
 
 #define PARTS_RKIMG \
 	"uuid_disk=${uuid_gpt_disk};" \
