diff --git a/src/common/menu/menu.cpp b/src/common/menu/menu.cpp
index 4532dae1d..8c394dc79 100644
--- a/src/common/menu/menu.cpp
+++ b/src/common/menu/menu.cpp
@@ -702,7 +702,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Back;
 				break;
 
-			case KEY_JOY4:
+			case KEY_JOY3:
 			case KEY_PAD_X:
 				mkey = MKEY_Clear;
 				break;
@@ -721,7 +721,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_UP:
 			case KEY_JOYAXIS2MINUS:
 			case KEY_JOYPOV1_UP:
-			case KEY_JOY9:
+			case KEY_JOY14:
 				mkey = MKEY_Up;
 				break;
 
@@ -729,7 +729,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_DOWN:
 			case KEY_JOYAXIS2PLUS:
 			case KEY_JOYPOV1_DOWN:
-			case KEY_JOY10:
+			case KEY_JOY15:
 				mkey = MKEY_Down;
 				break;
 
@@ -737,7 +737,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_LEFT:
 			case KEY_JOYAXIS1MINUS:
 			case KEY_JOYPOV1_LEFT:
-			case KEY_JOY11:
+			case KEY_JOY16:
 				mkey = MKEY_Left;
 				break;
 
@@ -745,7 +745,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_RIGHT:
 			case KEY_JOYAXIS1PLUS:
 			case KEY_JOYPOV1_RIGHT:
-			case KEY_JOY12:
+			case KEY_JOY17:
 				mkey = MKEY_Right;
 				break;
 			}
