diff --git a/Makefile b/Makefile
index 3e7941c..ceb21c3 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,6 @@ SOURCES := \
   src/*/*/*.cpp
 
 CPUFLAGS := $(MAKE_CPUFLAGS)
-STRIPCMD := strip --strip-all
 ifneq ($(ISWIN),)
   OUTNAME := dosbox_pure_libretro.dll
   CXX     ?= g++
@@ -58,7 +57,6 @@ else ifneq (,$(findstring ios,$(platform)))
     MINVERSION = -miphoneos-version-min=5.0
   endif
   COMMONFLAGS += $(MINVERSION) -Wno-ignored-optimization-argument -Wno-unknown-warning-option
-  STRIPCMD := strip -xS
 else ifeq ($(platform),tvos-arm64)
   ifeq ($(IOSSDK),)
     IOSSDK := $(shell xcodebuild -version -sdk appletvos Path)
@@ -67,7 +65,6 @@ else ifeq ($(platform),tvos-arm64)
   CXX     = c++ -arch arm64 -isysroot $(IOSSDK)
   LDFLAGS := -Wl,-dead_strip
   COMMONFLAGS += -DDISABLE_DYNAREC=1 -Wno-unknown-warning-option
-  STRIPCMD := strip -xS
 else ifneq ($(ISMAC),)
   OUTNAME := dosbox_pure_libretro.dylib
   CXX     ?= c++
@@ -81,7 +78,6 @@ else ifneq ($(ISMAC),)
   endif
   COMMONFLAGS  += $(ARCHFLAGS)
   LDFLAGS      += $(ARCHFLAGS)
-  STRIPCMD := strip -xS
 else ifeq ($(platform),windows) # For MSYS2 only
   OUTNAME := dosbox_pure_libretro.dll
   CXX     ?= g++
@@ -138,7 +134,6 @@ else ifeq ($(platform),gcw0)
   LDFLAGS := -Wl,--gc-sections -fno-ident
   CPUFLAGS := -ffast-math -march=mips32r2 -mtune=mips32r2 -mhard-float -fexpensive-optimizations -frename-registers
   COMMONFLAGS += -pthread
-  STRIPCMD := /opt/gcw0-toolchain/usr/mipsel-gcw0-linux-uclibc/bin/strip --strip-all
 else ifneq ($(findstring Haiku,$(shell uname -s)),)
   OUTNAME := dosbox_pure_libretro.so
   LDFLAGS := -Wl,--gc-sections -fno-ident -lroot -lnetwork
@@ -245,9 +240,6 @@ ifeq ($(STATIC_LINKING), 1)
 else
 	$(info Linking $@ ...)
 	$(CXX) $(LDFLAGS) -o $@ $^ $(LDLIBS)
-ifneq ($(BUILD),DEBUG)
-	$(STRIPCMD) $@
-endif
 endif
 
 define COMPILE
