diff --git a/src/doomdef.h b/src/doomdef.h
index 34fad7318f..06e77b0dd3 100644
--- a/src/doomdef.h
+++ b/src/doomdef.h
@@ -191,6 +191,8 @@ enum ESkillLevels
 #define KEY_JOY13			(KEY_FIRSTJOYBUTTON+12)
 #define KEY_JOY14			(KEY_FIRSTJOYBUTTON+13)
 #define KEY_JOY15			(KEY_FIRSTJOYBUTTON+14)
+#define KEY_JOY16			(KEY_FIRSTJOYBUTTON+15)
+#define KEY_JOY17			(KEY_FIRSTJOYBUTTON+16)        
 #define KEY_LASTJOYBUTTON		0x187
 #define KEY_JOYPOV1_UP			0x188
 #define KEY_JOYPOV1_RIGHT		0x189
diff --git a/src/menu/menu.cpp b/src/menu/menu.cpp
index 2af6616fa0..cf6ff999d1 100644
--- a/src/menu/menu.cpp
+++ b/src/menu/menu.cpp
@@ -666,17 +666,17 @@ bool M_Responder (event_t *ev)
 			ch = ev->data1;
 			switch (ch)
 			{
-			case KEY_JOY2:
+			case KEY_JOY1: //B button on rg552/rg503
 			case KEY_PAD_A:
 				mkey = MKEY_Back;
 				break;
 
-			case KEY_JOY1:
+			case KEY_JOY2: //A button on rg552/rg503
 			case KEY_PAD_B:
 				mkey = MKEY_Enter;
 				break;
 
-			case KEY_JOY4:
+			case KEY_JOY3: //X button on rg552/rg503
 			case KEY_PAD_X:
 				mkey = MKEY_Clear;
 				break;
@@ -695,7 +695,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_UP:
 			case KEY_JOYAXIS2MINUS:
 			case KEY_JOYPOV1_UP:
-			case KEY_JOY9:
+			case KEY_JOY14: //Dpad up on rg552/rg503
 				mkey = MKEY_Up;
 				break;
 
@@ -703,7 +703,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_DOWN:
 			case KEY_JOYAXIS2PLUS:
 			case KEY_JOYPOV1_DOWN:
-			case KEY_JOY10:
+			case KEY_JOY15: //Dpad down on rg552/rg503
 				mkey = MKEY_Down;
 				break;
 
@@ -711,7 +711,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_LEFT:
 			case KEY_JOYAXIS1MINUS:
 			case KEY_JOYPOV1_LEFT:
-			case KEY_JOY11:
+			case KEY_JOY16: //Dpad left on rg552/rg503
 				mkey = MKEY_Left;
 				break;
 
@@ -719,7 +719,7 @@ bool M_Responder (event_t *ev)
 			case KEY_PAD_LTHUMB_RIGHT:
 			case KEY_JOYAXIS1PLUS:
 			case KEY_JOYPOV1_RIGHT:
-			case KEY_JOY12:
+			case KEY_JOY17: //Dpad right on rg552/rg503
 				mkey = MKEY_Right;
 				break;
 			}
