diff --git a/es-core/src/components/BrightnessInfoComponent.h b/es-core/src/components/BrightnessInfoComponent.h
index 5faa179..59f780a 100644
--- a/es-core/src/components/BrightnessInfoComponent.h
+++ b/es-core/src/components/BrightnessInfoComponent.h
@@ -17,7 +17,7 @@ public:
 	void render(const Transform4x4f& parentTrans) override;
 	void update(int deltaTime) override;
 
-	void reset() { mBrightness = 1; }
+	void reset() { mBrightness = -1; }
 
 private:
 	NinePatchComponent* mFrame;
