--- GuiMenu.orig.cpp	2023-11-15 15:15:11.933455034 +0900
+++ GuiMenu.cpp	2023-11-15 15:12:18.226381068 +0900
@@ -2005,6 +2005,21 @@ void GuiMenu::openGamesSettings_batocera
 		if (SystemData::isNetplayActivated() && ApiSystem::getInstance()->isScriptingSupported(ApiSystem::NETPLAY))
 			s->addEntry(_("NETPLAY SETTINGS"), true, [this] { openNetplaySettings(); }, "iconNetplay");
 
+		// PPSSPP font change
+		auto pspfont_enabled = std::make_shared<SwitchComponent>(mWindow);
+		bool pfbaseEnabled = SystemConf::getInstance()->get("global.pspfont.enabled") == "1";
+		pspfont_enabled->setState(pfbaseEnabled);
+		s->addWithLabel(_("PPSSPP KOREAN FONT"), pspfont_enabled);
+		s->addSaveFunc([pspfont_enabled] {
+			if (pspfont_enabled->getState() == false) {
+				runSystemCommand("/usr/bin/ppsspp_font.sh", "", nullptr);
+			} else {
+				runSystemCommand("/usr/bin/ppsspp_font.sh enabled", "", nullptr);
+			}
+			bool pspfontenabled = pspfont_enabled->getState();
+			SystemConf::getInstance()->set("global.pspfont.enabled", pspfontenabled ? "1" : "0");
+			SystemConf::getInstance()->saveSystemConf();
+		});
 	}
 
 	mWindow->pushGui(s);
