diff -rupN mupen64plus.orig/Makefile mupen64plus/Makefile
--- mupen64plus.orig/Makefile	2022-02-07 10:11:16.000000000 -0500
+++ mupen64plus/Makefile	2022-03-26 18:03:43.323763489 -0400
@@ -304,7 +304,7 @@ else ifneq (,$(findstring AMLG,$(platfor
   
    HAVE_NEON = 1
    WITH_DYNAREC=arm
-   COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
+   #COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
 
 # Amlogic S905/S912
@@ -325,12 +325,10 @@ else ifneq (,$(findstring RK,$(platform)
    LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
 
    ifneq (,$(findstring RK33,$(platform)))
-      CPUFLAGS += -march=armv8-a+crc -mfloat-abi=hard -mfpu=neon-fp-armv8
       ifneq (,$(findstring RK3399,$(platform)))
-         CPUFLAGS += -mtune=cortex-a72.cortex-a53
-         GLES3 = 1
-      else ifneq (,$(findstring RK3328,$(platform)))
-         CPUFLAGS += -mtune=cortex-a53
+         CPUFLAGS += -march=armv8-a -mtune=cortex-a72.cortex-a53
+      else ifneq (,$(findstring RK3326,$(platform)))
+         CPUFLAGS += -mtune=cortex-a35
          GLES = 1
       endif
    else ifneq (,$(findstring RK3288,$(platform)))
@@ -342,9 +340,10 @@ else ifneq (,$(findstring RK,$(platform)
       COREFLAGS += -DEGL_NO_X11
    endif
 
+   GLES = 1
    GL_LIB := -lGLESv2
-   HAVE_NEON = 1
-   WITH_DYNAREC=arm
+   HAVE_NEON = 0
+   WITH_DYNAREC := aarch64
    COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
 
