diff --git a/Makefile b/Makefile
index 8a719d55..0872e619 100644
--- a/Makefile
+++ b/Makefile
@@ -60,6 +60,8 @@ else ifneq (,$(findstring rpi,$(platform)))
    override platform += unix
 else ifneq (,$(findstring odroid,$(platform)))
    override platform += unix
+else ifneq (,$(findstring @DEVICE@,$(platform)))
+   override platform += unix
 endif
 
 # system platform
@@ -171,6 +173,15 @@ ifneq (,$(findstring unix,$(platform)))
 
    endif
    
+  #@DEVICE@
+  ifneq (,$(findstring @DEVICE@,$(platform)))
+    TARGET := $(TARGET_NAME)_libretro.so
+    CPUFLAGS += -march=armv8-a+crc+simd -mtune=@TARGET_CPU@ -DARM_FIX -DCLASSIC -DARM64
+    WITH_DYNAREC = aarch64
+    GLES = 1
+    GL_LIB := -lGLESv2
+  endif
+
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD ?= $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
