diff --git a/libretro/libretro.c b/libretro/libretro.c
index eea606f5..8c244e1f 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -397,10 +397,7 @@ static void setup_variables(void)
          "Send audio lists to HLE RSP; disabled|enabled" },
       { "parallel-n64-gfxplugin",
          "GFX Plugin; auto"
-#if defined(HAVE_OPENGL) || defined(HAVE_OPENGLES)
          "|glide64|gln64|rice"
-#endif         
-         "|angrylion"
 #ifdef HAVE_PARALLEL
             "|parallel"
 #endif
@@ -410,41 +407,22 @@ static void setup_variables(void)
 #if defined(HAVE_OPENGL) || defined(HAVE_OPENGLES)
 		 "|hle"
 #endif
-		 "|cxd4"
 #ifdef HAVE_PARALLEL_RSP
          "|parallel"
 #endif
          },
-#if defined(HAVE_OPENGL) || defined(HAVE_OPENGLES) || defined(HAVE_PARALLEL)
       { "parallel-n64-screensize",
-#ifdef CLASSIC
-         "Resolution (restart); 320x240|640x480|960x720|1280x960|1440x1080|1600x1200|1920x1440|2240x1680|2880x2160|5760x4320" },
-#else
-         "Resolution (restart); 640x480|960x720|1280x960|1440x1080|1600x1200|1920x1440|2240x1680|2880x2160|5760x4320|320x240" },	
-#endif
+         "Resolution (restart); 320x240|640x480|960x720|1440x1080" },
       { "parallel-n64-aspectratiohint",
          "Aspect ratio hint (reinit); normal|widescreen" },
       { "parallel-n64-filtering",
 		 "(Glide64) Texture Filtering; automatic|N64 3-point|bilinear|nearest" },
-      { "parallel-n64-dithering",
-		 "(Angrylion) Dithering; enabled|disabled" },
       { "parallel-n64-polyoffset-factor",
        "(Glide64) Polygon Offset Factor; -3.0|-2.5|-2.0|-1.5|-1.0|-0.5|0.0|0.5|1.0|1.5|2.0|2.5|3.0|3.5|4.0|4.5|5.0|-3.5|-4.0|-4.5|-5.0"
       },
       { "parallel-n64-polyoffset-units",
        "(Glide64) Polygon Offset Units; -3.0|-2.5|-2.0|-1.5|-1.0|-0.5|0.0|0.5|1.0|1.5|2.0|2.5|3.0|3.5|4.0|4.5|5.0|-3.5|-4.0|-4.5|-5.0"
       },
-#endif
-      { "parallel-n64-angrylion-vioverlay",
-       "(Angrylion) VI Overlay; Filtered|AA+Blur|AA+Dedither|AA only|Unfiltered|Depth|Coverage"
-      },
-      { "parallel-n64-angrylion-sync",
-       "(Angrylion) Thread sync level; Low|Medium|High"
-      },
-       { "parallel-n64-angrylion-multithread",
-         "(Angrylion) Multi-threading; all threads|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31|32|33|34|35|36|37|38|39|40|41|42|43|44|45|46|47|48|49|50|51|52|53|54|55|56|57|58|59|60|61|62|63" },
-       { "parallel-n64-angrylion-overscan",
-         "(Angrylion) Hide overscan; disabled|enabled" },
       { "parallel-n64-virefresh",
          "VI Refresh (Overclock); auto|1500|2200" },
       { "parallel-n64-bufferswap",
