diff --git a/source/common/console/keydef.h b/source/common/console/keydef.h
index 108806774..deeb90de8 100644
--- a/source/common/console/keydef.h
+++ b/source/common/console/keydef.h
@@ -68,15 +68,17 @@ enum EKeyCodes
 
 	KEY_FIRSTJOYBUTTON		= 0x108,
 	KEY_JOY1				= KEY_FIRSTJOYBUTTON+0,
-	KEY_JOY2,
-	KEY_JOY3,
-	KEY_JOY4,
-	KEY_JOY5,
-	KEY_JOY6,
-	KEY_JOY7,
-	KEY_JOY8,
-	KEY_JOY14			= KEY_FIRSTJOYBUTTON+13,
-	KEY_JOY15			= KEY_FIRSTJOYBUTTON+14,
+	KEY_JOY2				= KEY_FIRSTJOYBUTTON+1,
+	KEY_JOY3				= KEY_FIRSTJOYBUTTON+2,
+	KEY_JOY4				= KEY_FIRSTJOYBUTTON+3,
+	KEY_JOY5				= KEY_FIRSTJOYBUTTON+4,
+	KEY_JOY6				= KEY_FIRSTJOYBUTTON+5,
+	KEY_JOY7				= KEY_FIRSTJOYBUTTON+6,
+	KEY_JOY8				= KEY_FIRSTJOYBUTTON+7,
+	KEY_JOY14				= KEY_FIRSTJOYBUTTON+13,
+	KEY_JOY15				= KEY_FIRSTJOYBUTTON+14,
+	KEY_JOY16				= KEY_FIRSTJOYBUTTON+15,
+	KEY_JOY17				= KEY_FIRSTJOYBUTTON+16,
 	KEY_LASTJOYBUTTON		= 0x187,
 	KEY_JOYPOV1_UP			= 0x188,
 	KEY_JOYPOV1_RIGHT		= 0x189,
diff --git a/source/common/menu/menu.cpp b/source/common/menu/menu.cpp
index 89017d67b..fcd09ffb8 100644
--- a/source/common/menu/menu.cpp
+++ b/source/common/menu/menu.cpp
@@ -692,20 +692,18 @@ bool M_Responder (event_t *ev)
 			ch = ev->data1;
 			switch (ch)
 			{
-			case KEY_JOY1:
-			case KEY_JOY3:
-			case KEY_JOY15:
+			case KEY_JOY2:
+			case KEY_JOY4:
 			case KEY_PAD_A:
 				mkey = MKEY_Enter;
 				break;
 
-			case KEY_JOY2:
-			case KEY_JOY14:
+			case KEY_JOY1:
 			case KEY_PAD_B:
 				mkey = MKEY_Back;
 				break;
 
-			case KEY_JOY4:
+			case KEY_JOY3:
 			case KEY_PAD_X:
 				mkey = MKEY_Clear;
 				break;
@@ -720,6 +718,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_PageDown;
 				break;
 
+			case KEY_JOY14:
 			case KEY_PAD_DPAD_UP:
 			case KEY_PAD_LTHUMB_UP:
 			case KEY_JOYAXIS2MINUS:
@@ -727,6 +726,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Up;
 				break;
 
+			case KEY_JOY15:
 			case KEY_PAD_DPAD_DOWN:
 			case KEY_PAD_LTHUMB_DOWN:
 			case KEY_JOYAXIS2PLUS:
@@ -734,6 +734,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Down;
 				break;
 
+			case KEY_JOY16:
 			case KEY_PAD_DPAD_LEFT:
 			case KEY_PAD_LTHUMB_LEFT:
 			case KEY_JOYAXIS1MINUS:
@@ -741,6 +742,7 @@ bool M_Responder (event_t *ev)
 				mkey = MKEY_Left;
 				break;
 
+			case KEY_JOY17:
 			case KEY_PAD_DPAD_RIGHT:
 			case KEY_PAD_LTHUMB_RIGHT:
 			case KEY_JOYAXIS1PLUS:
