diff -rupN emulationstation.orig/es-core/src/VolumeControl.cpp emulationstation/es-core/src/VolumeControl.cpp
--- emulationstation.orig/es-core/src/VolumeControl.cpp	2022-02-10 07:24:07.296897974 -0500
+++ emulationstation/es-core/src/VolumeControl.cpp	2022-02-10 07:24:26.744689725 -0500
@@ -213,7 +213,7 @@ static PulseAudioControl PulseAudio;
 #endif
 
 #if defined(__linux__)
-	std::string VolumeControl::mixerName = "Master";
+	std::string VolumeControl::mixerName = "Playback";
 	std::string VolumeControl::mixerCard = "default";
 #endif
 
