diff -rupN pcsx_rearmed.orig/Makefile.libretro pcsx_rearmed/Makefile.libretro
--- pcsx_rearmed.orig/Makefile.libretro 2022-04-03 01:27:52.509284300 +0200
+++ pcsx_rearmed/Makefile.libretro      2022-04-03 01:28:38.192433888 +0200
@@ -363,6 +363,17 @@ else ifeq ($(platform), rpi4_64)
         fpic := -fPIC
         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72 -ftree-vectorize

+# RG552 (32 bit)
+else ifeq ($(platform), RG552)
+    TARGET := $(TARGET_NAME)_libretro.so
+    fpic := -fPIC
+    CFLAGS += -march=armv8-a -marm -mfpu=neon-fp-armv8 -mfloat-abi=hard
+    HAVE_NEON = 1
+    ARCH = arm
+    BUILTIN_GPU = neon
+    DYNAREC = ari64
+    LDFLAGS += -static-libgcc -static-libstdc++ -fPIC
+
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
