diff --git a/es-app/src/guis/GuiMenu.cpp b/es-app/src/guis/GuiMenu.cpp
index 1865857..760094c 100644
--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -140,7 +140,7 @@ GuiMenu::GuiMenu(Window *window, bool animate) : GuiComponent(window), mMenu(win
 
 		if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::WIFI)) {
 			addEntry(_("NETWORK SETTINGS").c_str(), true, [this] { openNetworkSettings_batocera(); }, "iconNetwork");
-#if defined(AMD64) || defined(RK3326) || defined(RK3566) || defined(RK3566_X55) || defined(RK3588) || defined(RK3588_ACE) ||defined(RK3399)
+#if defined(AMD64) || defined(RK3326) || defined(RK3566) || defined(RK3566_X55) || defined(RK3566_BSP) || defined(RK3566_BSP_X55) || defined(RK3588) || defined(RK3588_ACE) ||defined(RK3399)
 		  addEntry(_("MOONLIGHT GAME STREAMING").c_str(), true, [this] { GuiMoonlight::show(mWindow); }, "iconGames");
 #endif
 		}
@@ -880,7 +880,7 @@ void GuiMenu::openSystemSettings_batocera()
 
 		s->addWithLabel(_("BRIGHTNESS"), brightnessComponent);
 
-#if defined(RK3566) || defined(RK3566_X55)
+#if defined(RK3566) || defined(RK3566_X55) || defined(RK3566_BSP) || defined(RK3566_BSP_X55)
 
                 // gamma
                 auto gamma = std::make_shared<SliderComponent>(mWindow, 1.f, 100.f, 1.f, "%");
@@ -1375,7 +1375,7 @@ void GuiMenu::openSystemSettings_batocera()
 	}
 
 // Do not show on S922X devices yet.
-#if defined(AMD64) || defined(RK3326) || defined(RK3566) || defined(RK3566_X55) || defined(RK3588) || defined(RK3588_ACE) || defined(RK3399)
+#if defined(AMD64) || defined(RK3326) || defined(RK3566) || defined(RK3566_X55) || defined(RK3566_BSP) || defined(RK3566_BSP_X55) || defined(RK3588) || defined(RK3588_ACE) || defined(RK3399)
         // Allow user control over how the device sleeps
         s->addGroup(_("HARDWARE / SUSPEND"));
         auto optionsSleep = std::make_shared<OptionListComponent<std::string> >(mWindow, _("DEVICE SUSPEND MODE"), false);
diff --git a/es-core/src/platform.cpp b/es-core/src/platform.cpp
index c0e6606..7832509 100644
--- a/es-core/src/platform.cpp
+++ b/es-core/src/platform.cpp
@@ -368,10 +368,18 @@ std::string getArchString()
 	return "rk3566";
 #endif
 
+#if RK3566_BSP
+        return "rk3566-BSP";
+#endif
+
 #if RK3566_X55
 	return "rk3566-x55";
 #endif
 
+#if RK3566_X55_BSP
+        return "rk3566-BSP-x55";
+#endif
+
 #if RK3588
 	return "rk3588";
 #endif
