diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-powkiddy-rgb10-max-3.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-powkiddy-rgb10-max-3.dts
new file mode 100644
index 000000000000..df5dde6eec79
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-powkiddy-rgb10-max-3.dts
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2023 JELOS (https://github.com/JustEnoughLinuxOS)
+ */
+
+/dts-v1/;
+
+#include "meson-g12b-odroid-go-ultra.dts"
+
+/ {
+	compatible = "powkiddy,rgb10-max-3", "amlogic,a311d", "amlogic,g12b";
+	model = "Powkiddy RGB10 MAX 3";
+
+	/delete-node/ hp-detect-switch;
+
+	joypad: gou_joypad {
+		button-adc-deadzone = <400>;
+		button-adc-fuzz = <64>;
+		sw12 {
+			gpios = <&gpio GPIOX_17 GPIO_ACTIVE_LOW>;
+		};
+		sw13 {
+			gpios = <&gpio GPIOX_16 GPIO_ACTIVE_LOW>;
+		};
+		sw14 {
+			gpios = <&gpio GPIOX_11 GPIO_ACTIVE_LOW>;
+		};
+		sw17 {
+			gpios = <&gpio GPIOX_13 GPIO_ACTIVE_LOW>;
+		};
+		sw18 {
+			gpios = <&gpio GPIOX_12 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+        sound {
+		model = "RGB10-MAX3";
+	};
+};
