diff -rupN linux.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c linux/drivers/gpu/drm/drm_panel_orientation_quirks.c
--- linux.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c	2023-09-12 12:02:57.953624819 +0000
+++ linux/drivers/gpu/drm/drm_panel_orientation_quirks.c	2023-09-12 12:03:27.490291628 +0000
@@ -477,7 +477,7 @@ EXPORT_SYMBOL(drm_get_panel_orientation_
 /* There are no quirks for non x86 devices yet */
 int drm_get_panel_orientation_quirk(int width, int height)
 {
-	return DRM_MODE_PANEL_ORIENTATION_UNKNOWN;
+	return DRM_MODE_PANEL_ORIENTATION_LEFT_UP;
 }
 EXPORT_SYMBOL(drm_get_panel_orientation_quirk);
 
