diff -rupN linux.orig/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi linux/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
--- linux.orig/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2023-09-12 12:02:56.937601871 +0000
+++ linux/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2023-09-12 12:03:27.490291628 +0000
@@ -61,18 +61,6 @@
 	gpu_opp_table: opp-table-gpu {
 		compatible = "operating-points-v2";
 
-		opp-124999998 {
-			opp-hz = /bits/ 64 <124999998>;
-			opp-microvolt = <800000>;
-		};
-		opp-249999996 {
-			opp-hz = /bits/ 64 <249999996>;
-			opp-microvolt = <800000>;
-		};
-		opp-285714281 {
-			opp-hz = /bits/ 64 <285714281>;
-			opp-microvolt = <800000>;
-		};
 		opp-399999994 {
 			opp-hz = /bits/ 64 <399999994>;
 			opp-microvolt = <800000>;
diff -rupN linux.orig/arch/arm64/boot/dts/amlogic/meson-g12b-a311d.dtsi linux/arch/arm64/boot/dts/amlogic/meson-g12b-a311d.dtsi
--- linux.orig/arch/arm64/boot/dts/amlogic/meson-g12b-a311d.dtsi	2023-09-12 12:02:56.937601871 +0000
+++ linux/arch/arm64/boot/dts/amlogic/meson-g12b-a311d.dtsi	2023-09-12 12:03:27.490291628 +0000
@@ -11,6 +11,11 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-667000000 {
+			opp-hz = /bits/ 64 <667000000>;
+			opp-microvolt = <731000>;
+		};
+
 		opp-1000000000 {
 			opp-hz = /bits/ 64 <1000000000>;
 			opp-microvolt = <761000>;
@@ -51,6 +56,11 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-667000000 {
+			opp-hz = /bits/ 64 <667000000>;
+			opp-microvolt = <731000>;
+		};
+
 		opp-1000000000 {
 			opp-hz = /bits/ 64 <1000000000>;
 			opp-microvolt = <731000>;
diff -rupN linux.orig/arch/arm64/boot/dts/amlogic/meson-g12b-s922x.dtsi linux/arch/arm64/boot/dts/amlogic/meson-g12b-s922x.dtsi
--- linux.orig/arch/arm64/boot/dts/amlogic/meson-g12b-s922x.dtsi	2023-09-12 12:02:56.937601871 +0000
+++ linux/arch/arm64/boot/dts/amlogic/meson-g12b-s922x.dtsi	2023-09-12 12:03:27.490291628 +0000
@@ -11,6 +11,11 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-667000000 {
+			opp-hz = /bits/ 64 <667000000>;
+			opp-microvolt = <731000>;
+		};
+
 		opp-1000000000 {
 			opp-hz = /bits/ 64 <1000000000>;
 			opp-microvolt = <731000>;
@@ -56,6 +61,11 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-667000000 {
+			opp-hz = /bits/ 64 <667000000>;
+			opp-microvolt = <751000>;
+		};
+
 		opp-1000000000 {
 			opp-hz = /bits/ 64 <1000000000>;
 			opp-microvolt = <771000>;
