diff -rupN mame.orig/src/lib/netlist/plib/ppmf.h mame/src/lib/netlist/plib/ppmf.h
--- mame.orig/src/lib/netlist/plib/ppmf.h	2022-07-04 11:02:16.563354243 -0400
+++ mame/src/lib/netlist/plib/ppmf.h	2022-07-04 11:03:01.483272032 -0400
@@ -81,6 +81,7 @@ namespace plib {
 #include "ptypes.h"
 
 #include <algorithm>
+#include <cstddef> // ptrdiff_t
 #include <cstdint> // uintptr_t
 #include <type_traits>
 #include <utility>
