diff --git a/hal/rtl8821c/rtl8821c_ops.c b/hal/rtl8821c/rtl8821c_ops.c
index 7461142..fffd472 100755
--- a/hal/rtl8821c/rtl8821c_ops.c
+++ b/hal/rtl8821c/rtl8821c_ops.c
@@ -524,7 +524,7 @@ u8 rtl8821c_read_efuse(PADAPTER adapter)
 
 	if (hal_read_mac_hidden_rpt(adapter) != _SUCCESS)
 		goto exit;
-	{
+	/*{
 		struct hal_spec_t *hal_spec = GET_HAL_SPEC(adapter);
 
 		if (hal_spec->hci_type <= 3 && hal_spec->hci_type >= 1) {
@@ -532,7 +532,7 @@ u8 rtl8821c_read_efuse(PADAPTER adapter)
 			RTW_INFO("EEPROM Disable BT-coex by hal_spec\n");
 			rtw_btcoex_wifionly_AntInfoSetting(adapter);
 		}
-	}
+	}*/
 
 	rtw_phydm_read_efuse(adapter);

