--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts	2024-03-15 18:26:31.236348216 +0300
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts	2024-03-15 18:26:48.664293252 +0300
@@ -139,7 +139,7 @@
 
 &gpio_keys_control {
 	button-a {
-		gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_LOW>;
+		gpios = <&gpio3 RK_PC2 GPIO_ACTIVE_LOW>;
 		label = "EAST";
 		linux,code = <BTN_EAST>;
 	};
@@ -175,7 +175,7 @@
 	};
 
 	button-y {
-		gpios = <&gpio3 RK_PC2 GPIO_ACTIVE_LOW>;
+		gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_LOW>;
 		label = "WEST";
 		linux,code = <BTN_WEST>;
 	};
