diff -rupN nvtop.orig/CMakeLists.txt nvtop/CMakeLists.txt
--- nvtop.orig/CMakeLists.txt	2024-01-13 18:01:55.173963740 +0000
+++ nvtop/CMakeLists.txt	2024-01-14 06:32:20.456483176 +0000
@@ -45,7 +45,7 @@ set_property(TARGET ncurses PROPERTY
 #///////////////////////////////////////////////////////////////////#
 
 # Use full RPATH on build tree
-set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_SKIP_BUILD_RPATH TRUE)
 # Do not build with install RPATH
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 # Set the RPATH when install
@@ -70,12 +70,12 @@ if(APPLE)
   set(PANTHOR_SUPPORT_DEFAULT OFF)
 else()
   set(APPLE_SUPPORT_DEFAULT OFF)
-  set(NVIDIA_SUPPORT_DEFAULT ON)
-  set(AMDGPU_SUPPORT_DEFAULT ON)
-  set(INTEL_SUPPORT_DEFAULT ON)
-  set(MSM_SUPPORT_DEFAULT ON)
+  set(NVIDIA_SUPPORT_DEFAULT OFF)
+  set(AMDGPU_SUPPORT_DEFAULT OFF)
+  set(INTEL_SUPPORT_DEFAULT OFF)
+  set(MSM_SUPPORT_DEFAULT OFF)
   set(PANFROST_SUPPORT_DEFAULT ON)
-  set(PANTHOR_SUPPORT_DEFAULT ON)
+  set(PANTHOR_SUPPORT_DEFAULT OFF)
 endif()
 
 option(NVIDIA_SUPPORT "Build support for NVIDIA GPUs through libnvml" ${NVIDIA_SUPPORT_DEFAULT})
@@ -107,18 +107,18 @@ install(FILES
   DESTINATION share/man/man1/
   PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
   RENAME nvtop.1)
-install(FILES
-  "${CMAKE_CURRENT_SOURCE_DIR}/desktop/nvtop.svg"
-  DESTINATION share/icons
-  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
-install(FILES
-  "${CMAKE_CURRENT_SOURCE_DIR}/desktop/nvtop.desktop"
-  DESTINATION share/applications
-  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
-install(FILES
-  "${CMAKE_CURRENT_BINARY_DIR}/desktop/nvtop.metainfo.xml"
-  DESTINATION share/metainfo
-  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+#install(FILES
+#  "${CMAKE_CURRENT_SOURCE_DIR}/desktop/nvtop.svg"
+#  DESTINATION share/icons
+#  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+#install(FILES
+#  "${CMAKE_CURRENT_SOURCE_DIR}/desktop/nvtop.desktop"
+#  DESTINATION share/applications
+#  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+#install(FILES
+#  "${CMAKE_CURRENT_BINARY_DIR}/desktop/nvtop.metainfo.xml"
+#  DESTINATION share/metainfo
+#  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 
 configure_file(
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
diff -rupN nvtop.orig/src/panfrost_drm.h nvtop/src/panfrost_drm.h
--- nvtop.orig/src/panfrost_drm.h	2024-01-13 18:01:55.177963823 +0000
+++ nvtop/src/panfrost_drm.h	2024-01-14 06:15:20.775076824 +0000
@@ -6,7 +6,7 @@
 #ifndef _PANFROST_DRM_H_
 #define _PANFROST_DRM_H_
 
-#include "drm.h"
+//#include "drm.h"
 
 #if defined(__cplusplus)
 extern "C" {
