diff --git a/buildfiles/cmake/ConfigureCompiler.cmake b/buildfiles/cmake/ConfigureCompiler.cmake
index 1cf3840df..f42d6d8f1 100644
--- a/buildfiles/cmake/ConfigureCompiler.cmake
+++ b/buildfiles/cmake/ConfigureCompiler.cmake
@@ -43,12 +43,10 @@ else()
 		endif()
 	endif()
 
-	add_compile_options(-Werror=old-style-cast)
 	add_compile_options(-Werror=sign-compare)
 	add_compile_options(-Werror=reorder)
 	add_compile_options(-Werror=return-type)
 	add_compile_options(-Werror=overloaded-virtual)
-	add_compile_options(-Werror=missing-noreturn)
 	add_compile_options(-Wunused-parameter)
 	add_compile_options(-Wignored-qualifiers)
 	add_compile_options(-Wredundant-move)
