diff -rupN wireguard-linux-compat.orig/src/compat/compat.h wireguard-linux-compat/src/compat/compat.h
--- wireguard-linux-compat.orig/src/compat/compat.h	2022-05-12 09:47:59.423533102 +0000
+++ wireguard-linux-compat/src/compat/compat.h	2022-05-12 09:49:46.623499934 +0000
@@ -686,14 +686,14 @@ struct __compat_dummy_container { char d
 #define genl_dump_check_consistent(a, b) genl_dump_check_consistent(a, b, &genl_family)
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0) && !defined(ISRHEL7)
+/* #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0) && !defined(ISRHEL7)
 static inline void *skb_put_data(struct sk_buff *skb, const void *data, unsigned int len)
 {
 	void *tmp = skb_put(skb, len);
 	memcpy(tmp, data, len);
 	return tmp;
 }
-#endif
+#endif */
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0) && !defined(ISRHEL7)
 #define napi_complete_done(n, work_done) napi_complete(n)
