diff -rupN retroarch.orig/menu/drivers/ozone.c retroarch/menu/drivers/ozone.c
--- retroarch.orig/menu/drivers/ozone.c	2022-03-12 06:46:06.484390460 -0500
+++ retroarch/menu/drivers/ozone.c	2022-03-12 06:46:56.572060794 -0500
@@ -7066,7 +7066,7 @@ static void *ozone_init(void **userdata,
    ozone->last_width        = width;
    ozone->last_height       = height;
    ozone->last_scale_factor = gfx_display_get_dpi_scale(p_disp,
-         settings, width, height, false, false);
+         settings, width, height, false, false) * 2.000f;
    ozone->last_thumbnail_scale_factor = settings->floats.ozone_thumbnail_scale_factor;
 
    file_list_initialize(&ozone->selection_buf_old);
@@ -8166,7 +8166,7 @@ static void ozone_render(void *data,
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    scale_factor = gfx_display_get_dpi_scale(p_disp, settings,
-         width, height, false, false);
+         width, height, false, false) * 2.000f;
    thumbnail_scale_factor = settings->floats.ozone_thumbnail_scale_factor;
 
    if ((scale_factor != ozone->last_scale_factor) ||
