diff -rupN retroarch.orig/menu/menu_setting.c retroarch/menu/menu_setting.c
--- retroarch.orig/menu/menu_setting.c	2022-03-05 15:34:24.206615672 -0500
+++ retroarch/menu/menu_setting.c	2022-03-05 20:09:51.435336191 -0500
@@ -9321,7 +9321,7 @@ static bool setting_append_list(
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_QUIT_RETROARCH,
-               MENU_ENUM_LABEL_VALUE_RESTART_RETROARCH,
+               MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
                &group_info,
                &subgroup_info,
                parent_group);
