diff -rupN retroarch.orig/frontend/drivers/platform_unix.c retroarch/frontend/drivers/platform_unix.c
--- retroarch.orig/frontend/drivers/platform_unix.c	2022-02-05 19:38:43.852018492 -0500
+++ retroarch/frontend/drivers/platform_unix.c	2022-02-05 20:10:30.735583922 -0500
@@ -2048,12 +2048,6 @@ static void frontend_unix_deinit(void *d
    android_app_destroy(android_app);
 #endif
 
-#ifdef HAVE_LAKKA
-   /* Reset brightness to maximum */
-   if (settings->uints.screen_brightness != DEFAULT_SCREEN_BRIGHTNESS)
-      frontend_unix_set_screen_brightness(DEFAULT_SCREEN_BRIGHTNESS);
-#endif
-
    frontend_unix_set_gamemode(false);
 }
 
@@ -2885,11 +2879,7 @@ frontend_ctx_driver_t frontend_ctx_unix
 #else
    NULL,                               /* get_lakka_version */
 #endif
-#if defined(HAVE_LAKKA_SWITCH) || (defined(HAVE_LAKKA) && defined(HAVE_ODROIDGO2))
-   frontend_unix_set_screen_brightness,/* set_screen_brightness */
-#else
-   NULL,                         /* set_screen_brightness */
-#endif
+   NULL,                         /* set_screen_brightness? No. */
    frontend_unix_watch_path_for_changes,
    frontend_unix_check_for_path_changes,
    frontend_unix_set_sustained_performance_mode,
