diff -rupN linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts
--- linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts	2023-10-03 23:48:15.144113656 +0000
+++ linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts	2023-10-03 23:48:55.102040451 +0000
@@ -8,7 +8,7 @@
 #include "rk3566-anbernic-rg353x.dtsi"
 
 / {
-	model = "RG353P";
+	model = "Anbernic RG353P";
 	compatible = "anbernic,rg353p", "rockchip,rk3566";
 
 	aliases {
diff -rupN linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts
--- linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts	2023-10-03 23:48:15.144113656 +0000
+++ linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts	2023-10-03 23:49:18.163158275 +0000
@@ -8,7 +8,7 @@
 #include "rk3566-anbernic-rg353x.dtsi"
 
 / {
-	model = "RG353PS";
+	model = "Anbernic RG353PS";
 	compatible = "anbernic,rg353ps", "rockchip,rk3566";
 
 	aliases {
diff -rupN linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353v.dts linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353v.dts
--- linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353v.dts	2023-10-03 23:48:15.144113656 +0000
+++ linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353v.dts	2023-10-03 23:49:09.254725978 +0000
@@ -8,7 +8,7 @@
 #include "rk3566-anbernic-rg353x.dtsi"
 
 / {
-	model = "RG353V";
+	model = "Anbernic RG353V";
 	compatible = "anbernic,rg353v", "rockchip,rk3566";
 
 	aliases {
diff -rupN linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353vs.dts linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353vs.dts
--- linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353vs.dts	2023-10-03 23:48:15.144113656 +0000
+++ linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353vs.dts	2023-10-03 23:49:30.947779716 +0000
@@ -8,7 +8,7 @@
 #include "rk3566-anbernic-rg353x.dtsi"
 
 / {
-	model = "RG353VS";
+	model = "Anbernic RG353VS";
 	compatible = "anbernic,rg353vs", "rockchip,rk3566";
 
 	aliases {
diff -rupN linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts
--- linux-6.6-rc3.orig/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts	2023-10-03 23:48:15.144113656 +0000
+++ linux-6.6-rc3/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts	2023-10-03 23:49:41.912313646 +0000
@@ -8,7 +8,7 @@
 #include "rk3566-anbernic-rgxx3.dtsi"
 
 / {
-	model = "RG503";
+	model = "Anbernic RG503";
 	compatible = "anbernic,rg503", "rockchip,rk3566";
 
 	aliases {
