diff --git a/mupen64plus-core/src/r4300/r4300.c b/mupen64plus-core/src/r4300/r4300.c
index 744198d6..73cb1f27 100644
--- a/mupen64plus-core/src/r4300/r4300.c
+++ b/mupen64plus-core/src/r4300/r4300.c
@@ -50,8 +50,8 @@
 unsigned int r4300emu = 0;
 unsigned int count_per_op = COUNT_PER_OP_DEFAULT;
 unsigned int llbit;
-int stop;
 #if NEW_DYNAREC < NEW_DYNAREC_ARM
+int stop;
 int64_t reg[32], hi, lo;
 uint32_t next_interrupt;
 struct precomp_instr *PC;
diff --git a/mupen64plus-rsp-paraLLEl/rsp_disasm.cpp b/mupen64plus-rsp-paraLLEl/rsp_disasm.cpp
index 9b1f6eda..bbb84e74 100644
--- a/mupen64plus-rsp-paraLLEl/rsp_disasm.cpp
+++ b/mupen64plus-rsp-paraLLEl/rsp_disasm.cpp
@@ -1,5 +1,6 @@
 #include "rsp_disasm.hpp"
 #include <assert.h>
+#include <cstdint>
 
 namespace RSP
 {
@@ -582,4 +583,4 @@ std::string disassemble(uint32_t pc, uint32_t instr)
 
 	return "???";
 }
-}
\ No newline at end of file
+}
