diff -rupN grub.orig/configure.ac grub/configure.ac
--- grub.orig/configure.ac	2023-08-23 15:20:39.432750326 +0000
+++ grub/configure.ac	2023-08-23 15:57:20.407704581 +0000
@@ -54,7 +54,7 @@ AC_CANONICAL_TARGET
 program_prefix="${save_program_prefix}"
 
 AM_INIT_AUTOMAKE([1.11])
-AC_PREREQ(2.64)
+AC_PREREQ([2.64])
 AC_CONFIG_SRCDIR([include/grub/dl.h])
 AC_CONFIG_HEADERS([config-util.h])

diff -rupN grub.orig/grub-core/osdep/unix/sleep.c grub/grub-core/osdep/unix/sleep.c
--- grub.orig/grub-core/osdep/unix/sleep.c      2023-08-23 15:20:39.480752766 +0000
+++ grub/grub-core/osdep/unix/sleep.c   2023-08-23 15:57:20.407704581 +0000
@@ -16,6 +16,8 @@
  *  along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
  */

+#include <config-util.h>
+#include <config.h>
 #include <time.h>
 #include <grub/time.h>
