diff --git a/core/NormalRga.h b/core/NormalRga.h
index 9ac6ee8..e04eef5 100755
--- a/core/NormalRga.h
+++ b/core/NormalRga.h
@@ -41,7 +41,7 @@
 #include <hardware/hardware.h>
 #endif
 
-#include "drmrga.h"
+#include "rga/drmrga.h"
 #include "rga.h"
 
 #include "NormalRgaContext.h"
diff --git a/include/GrallocOps.h b/include/GrallocOps.h
index 97e92d4..eed4e29 100755
--- a/include/GrallocOps.h
+++ b/include/GrallocOps.h
@@ -45,7 +45,7 @@
 #include <utils/Log.h>
 #include <log/log_main.h>
 
-#include "drmrga.h"
+#include "rga/drmrga.h"
 #include "rga.h"
 
 // -------------------------------------------------------------------------------
diff --git a/include/RgaApi.h b/include/RgaApi.h
index b0049f7..0440de0 100755
--- a/include/RgaApi.h
+++ b/include/RgaApi.h
@@ -30,7 +30,7 @@
 #include <sys/mman.h>
 #include <linux/stddef.h>
 
-#include "drmrga.h"
+#include "rga/drmrga.h"
 #include "rga.h"
 
 #ifdef __cplusplus
diff --git a/include/RockchipRga.h b/include/RockchipRga.h
index 082f9e1..4aca7f1 100755
--- a/include/RockchipRga.h
+++ b/include/RockchipRga.h
@@ -30,7 +30,7 @@
 #include <sys/mman.h>
 #include <linux/stddef.h>
 
-#include "drmrga.h"
+#include "rga/drmrga.h"
 #include "GrallocOps.h"
 #include "RgaUtils.h"
 #include "rga.h"
