diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index 9b4a20bd9..9a46971fd 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -993,15 +993,20 @@ void retro_init(void)
 
    retro_base_dir /= "PPSSPP";
 
-   g_Config.currentDirectory = retro_base_dir;
-   g_Config.defaultCurrentDirectory = retro_base_dir;
-   g_Config.memStickDirectory = retro_save_dir;
-   g_Config.flash0Directory = retro_base_dir / "flash0";
-   g_Config.internalDataDirectory = retro_base_dir;
+   g_Config.defaultCurrentDirectory = Path(getenv("HOME"));
+   g_Config.memStickDirectory = g_Config.defaultCurrentDirectory / ".config/ppsspp";
+   g_Config.flash0Directory = g_Config.memStickDirectory / "assets" / "flash0";
+   g_Config.internalDataDirectory = g_Config.memStickDirectory;
+
+   //g_Config.currentDirectory = retro_base_dir;
+   //g_Config.defaultCurrentDirectory = retro_base_dir;
+   //g_Config.memStickDirectory = retro_save_dir;
+   //g_Config.flash0Directory = retro_base_dir / "flash0";
+   //g_Config.internalDataDirectory = retro_base_dir;
    g_Config.bEnableNetworkChat = false;
    g_Config.bDiscordPresence = false;
 
-   VFSRegister("", new DirectoryAssetReader(retro_base_dir));
+   VFSRegister("", new DirectoryAssetReader(g_Config.memStickDirectory / "assets"));
 
    host = new LibretroHost();
 }
