diff -rupN linux-6.6.9.orig/drivers/hwmon/oxp-sensors.c linux-6.6.9/drivers/hwmon/oxp-sensors.c
--- linux-6.6.9.orig/drivers/hwmon/oxp-sensors.c	2024-01-04 00:22:05.661168063 +0000
+++ linux-6.6.9/drivers/hwmon/oxp-sensors.c	2024-01-04 11:17:07.777068733 +0000
@@ -42,6 +42,7 @@ static bool unlock_global_acpi_lock(void
 enum oxp_board {
 	aok_zoe_a1 = 1,
 	aya_neo_2,
+	aya_neo_2s,
 	aya_neo_air,
 	aya_neo_air_pro,
 	aya_neo_geek,
@@ -91,6 +92,13 @@ static const struct dmi_system_id dmi_ta
 		},
 		.driver_data = (void *)aya_neo_2,
 	},
+        {
+                .matches = {
+                        DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+                        DMI_EXACT_MATCH(DMI_BOARD_NAME, "AYANEO 2S"),
+                },
+                .driver_data = (void *)aya_neo_2s,
+        },
 	{
 		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
@@ -331,6 +339,7 @@ static int oxp_platform_read(struct devi
 				return ret;
 			switch (board) {
 			case aya_neo_2:
+			case aya_neo_2s:
 			case aya_neo_air:
 			case aya_neo_air_pro:
 			case aya_neo_geek:
