diff -rupN retroarch.orig/gfx/drivers/oga_gfx.c retroarch/gfx/drivers/oga_gfx.c
--- retroarch.orig/gfx/drivers/oga_gfx.c	2022-02-28 19:40:46.222185295 -0500
+++ retroarch/gfx/drivers/oga_gfx.c	2022-02-28 19:44:14.187909576 -0500
@@ -22,7 +22,7 @@
 #include "../../verbosity.h"
 #include <fcntl.h>
 #include <rga/RgaApi.h>
-#include <rga/RockchipRgaMacro.h>
+#include <rga/drmrga.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
 #include <drm/drm_fourcc.h>
