diff --git a/yabause/src/nx/main.cpp b/yabause/src/nx/main.cpp
index 7d7ee2a30..51f5c12ed 100644
--- a/yabause/src/nx/main.cpp
+++ b/yabause/src/nx/main.cpp
@@ -190,7 +190,7 @@ OSD_struct *OSDCoreList[] = {
 static SDL_Window* wnd;
 static SDL_GLContext glc;
 int g_EnagleFPS = 1;
-int g_resolution_mode = 0;
+int g_resolution_mode = 3;
 int g_keep_aspect_rate = 0;
 int g_scsp_sync = 1;
 int g_frame_skip = 0;
diff --git a/yabause/src/sdl/main.c b/yabause/src/sdl/main.c
index b990c1df5..8577c5546 100644
--- a/yabause/src/sdl/main.c
+++ b/yabause/src/sdl/main.c
@@ -141,7 +141,7 @@ NULL
 #endif
 
 static int fullscreen = 0;
-static int lowres_mode = 0;
+static int lowres_mode = 1;
 
 static char biospath[256] = "\0";
 static char strgsyslangeid[256] = "english";
