diff --git a/CMakeModules/FindLIBEVDEV.cmake b/CMakeModules/FindLIBEVDEV.cmake
index b560a27f..f2c8adce 100644
--- a/CMakeModules/FindLIBEVDEV.cmake
+++ b/CMakeModules/FindLIBEVDEV.cmake
@@ -11,8 +11,7 @@ FIND_PATH(
     LIBEVDEV_INCLUDE_DIR libevdev/libevdev.h
     HINTS ${PC_LIBEVDEV_INCLUDEDIR} ${PC_LIBEVDEV_INCLUDE_DIRS}
     /usr/include
-    /usr/local/include
-    /usr/local/include/libevdev-1.0
+    /usr/include/libevdev-1.0
     ${LIBEVDEV_PATH_INCLUDES}
 )
