diff --git a/src/common/log.h b/src/common/log.h
index 132b65d3..9e7064a8 100644
--- a/src/common/log.h
+++ b/src/common/log.h
@@ -1,6 +1,8 @@
 #pragma once
 #include "types.h"
 #include <cinttypes>
+#include <cstdint>
+#include <cstdarg>
 #include <mutex>
 
 enum LOGLEVEL
diff --git a/src/frontend-common/cubeb_audio_stream.cpp b/src/frontend-common/cubeb_audio_stream.cpp
index 099e62b5..7e1e0acc 100644
--- a/src/frontend-common/cubeb_audio_stream.cpp
+++ b/src/frontend-common/cubeb_audio_stream.cpp
@@ -3,6 +3,8 @@
 #include "common/log.h"
 Log_SetChannel(CubebAudioStream);
 
+#include <string>
+
 #ifdef _WIN32
 #include "common/windows_headers.h"
 #include <objbase.h>
