diff --git a/include/linux/compiler.h b/include/linux/compiler.h
index 219aa5ddbc73..35fc33d00985 100644
--- a/include/linux/compiler.h
+++ b/include/linux/compiler.h
@@ -213,6 +213,8 @@ void ftrace_likely_update(struct ftrace_likely_data *f, int val,
 #define function_nocfi(x) (x)
 #endif
 
+#include <asm/rwonce.h>
+
 #endif /* __KERNEL__ */
 
 /*
@@ -245,6 +247,4 @@ static inline void *offset_to_ptr(const int *off)
  */
 #define prevent_tail_call_optimization()	mb()
 
-#include <asm/rwonce.h>
-
 #endif /* __LINUX_COMPILER_H */
