diff -rupN flycast.orig/Makefile flycast/Makefile
--- flycast.orig/Makefile	2022-06-18 13:29:52.675787687 -0400
+++ flycast/Makefile	2022-06-18 13:36:32.802306197 -0400
@@ -403,8 +403,8 @@ else ifeq ($(platform), arm64_cortex_a53
 
 #######################################
 
-# ARM64 SM1 Odroid C4
-else ifeq ($(platform), odroidc4)
+# ARM64 RG503, RG353P
+else ifeq ($(platform), RK3566)
 	EXT ?= so
 	TARGET := $(TARGET_NAME)_libretro.$(EXT)
 	SHARED += -shared -Wl,--version-script=link.T
