diff -rupN wireplumber-0.4.15.orig/src/systemd/system/wireplumber.service.in wireplumber-0.4.15/src/systemd/system/wireplumber.service.in
--- wireplumber-0.4.15.orig/src/systemd/system/wireplumber.service.in	2023-11-13 16:40:22.585908363 +0000
+++ wireplumber-0.4.15/src/systemd/system/wireplumber.service.in	2023-11-13 16:42:13.896286333 +0000
@@ -14,9 +14,12 @@ Type=simple
 AmbientCapabilities=CAP_SYS_NICE
 ExecStart=@WP_BINARY@
 Restart=on-failure
-User=pipewire
+User=root
 Environment=PIPEWIRE_RUNTIME_DIR=%t/pipewire
 Environment=GIO_USE_VFS=local
+Environment=XDG_RUNTIME_DIR=/var/run/0-runtime-dir
+Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/run/dbus/system_bus_socket
+Environment=HOME=/storage
 
 [Install]
 WantedBy=pipewire.service
diff -rupN wireplumber-0.4.15.orig/src/systemd/system/wireplumber@.service.in wireplumber-0.4.15/src/systemd/system/wireplumber@.service.in
--- wireplumber-0.4.15.orig/src/systemd/system/wireplumber@.service.in	2023-11-13 16:40:22.585908363 +0000
+++ wireplumber-0.4.15/src/systemd/system/wireplumber@.service.in	2023-11-13 16:41:37.274187302 +0000
@@ -19,7 +19,7 @@ Type=simple
 AmbientCapabilities=CAP_SYS_NICE
 ExecStart=@WP_BINARY@ -c %i.conf
 Restart=on-failure
-User=pipewire
+User=root
 Environment=PIPEWIRE_RUNTIME_DIR=%t/pipewire
 Environment=GIO_USE_VFS=local
 
