diff --git a/src/GLideNHQ/TxCache.cpp b/src/GLideNHQ/TxCache.cpp
index 8bea6974..5af1741a 100644
--- a/src/GLideNHQ/TxCache.cpp
+++ b/src/GLideNHQ/TxCache.cpp
@@ -31,6 +31,7 @@
 #include <memory.h>
 #include <stdlib.h>
 #include <assert.h>
+#include <string.h>
 
 #include <osal_files.h>
 
diff --git a/src/GLideNHQ/TxReSample.cpp b/src/GLideNHQ/TxReSample.cpp
index 17132626..3fc1689b 100644
--- a/src/GLideNHQ/TxReSample.cpp
+++ b/src/GLideNHQ/TxReSample.cpp
@@ -25,6 +25,7 @@
 #include "TxDbg.h"
 #include <stdlib.h>
 #include <memory.h>
+#include <string.h>
 
 #define _USE_MATH_DEFINES
 #include <math.h>
