diff --git a/src/mupenplus/GLideN64_mupenplus.h b/src/mupenplus/GLideN64_mupenplus.h
index c925d152..d62fd237 100644
--- a/src/mupenplus/GLideN64_mupenplus.h
+++ b/src/mupenplus/GLideN64_mupenplus.h
@@ -1,9 +1,9 @@
-#ifndef GLIDEN64_MUPENPLUS_H
+#ifndef GLIDEN66_MUPENPLUS_H
 #define GLIDEN64_MUPENPLUS_H
 
-#include "m64p_common.h"
-#include "m64p_config.h"
-#include "m64p_vidext.h"
+#include "mupen64plus/src/api/m64p_common.h"
+#include "mupen64plus/src/api/m64p_config.h"
+#include "mupen64plus/src/api/m64p_vidext.h"
 
 #define PLUGIN_VERSION              0x020000
 #define VIDEO_PLUGIN_API_VERSION	0x020200
