diff --git a/gfx/drivers_context/drm_go2_ctx.c b/gfx/drivers_context/drm_go2_ctx.c
index 856aa0f9a2..baa0ca723a 100644
--- a/gfx/drivers_context/drm_go2_ctx.c
+++ b/gfx/drivers_context/drm_go2_ctx.c
@@ -122,7 +122,7 @@ static void *gfx_ctx_go2_drm_init(void *video_driver)
    drm->native_height = go2_display_width_get(drm->display);
 
    /* This driver should only be used on rotated screens */
-   if (drm->native_width < drm->native_height)
+   if (drm->native_width <= drm->native_height)
    {
       /* This should be fixed by using wayland/weston... */
       go2_display_destroy(drm->display);
