diff -rupN fileman.orig/src/mainWindow.cpp fileman/src/mainWindow.cpp
--- fileman.orig/src/mainWindow.cpp	2023-05-31 02:21:54.218866701 +0900
+++ fileman/src/mainWindow.cpp	2023-05-31 02:22:51.551941522 +0900
@@ -160,6 +160,12 @@ void MainWindow::render(const bool p_foc
 
 }
 
+// Supported extensions
+bool MainWindow::extensionIsSupported(const std::string &p_ext)
+{
+   return p_ext == "txt" || p_ext == "xml" || p_ext == "log" || p_ext == "cfg" || p_ext == "conf";
+}
+
 //------------------------------------------------------------------------------
 
 // Move cursor up / down
@@ -294,9 +300,12 @@ void MainWindow::openHighlightedFile(voi
    }
 
    // If the file is binary, do nothing
-   std::string filePath = m_title + (m_title == "/" ? "" : "/") + m_fileLister[m_cursor].m_name;
-   if (! (FileUtils::fileIsText(filePath) || m_fileLister[m_cursor].m_size == 0))
-      return;
+   if (! (MainWindow::extensionIsSupported(m_fileLister[m_cursor].m_ext)))
+   {
+      std::string filePath = m_title + (m_title == "/" ? "" : "/") + m_fileLister[m_cursor].m_name;
+      if (! (FileUtils::fileIsText(filePath) || m_fileLister[m_cursor].m_size == 0))
+         return;
+   }
 
    // Dialog 'view as text' / 'edit as text'
    int action = -1;
