diff --git a/simple64-gui/CMakeLists.txt b/simple64-gui/CMakeLists.txt
index a5d3ccb..8096e76 100644
--- a/simple64-gui/CMakeLists.txt
+++ b/simple64-gui/CMakeLists.txt
@@ -25,12 +25,12 @@ endif()
 find_package(SDL2 REQUIRED)
 include_directories(${SDL2_INCLUDE_DIRS})
 
-find_package(Qt6 REQUIRED COMPONENTS Widgets WebSockets)
+find_package(Qt5 5.15 REQUIRED COMPONENTS Widgets WebSockets)
 if (UNIX)
-    find_package(Qt6 REQUIRED COMPONENTS DBus)
+    find_package(Qt5 5.15 REQUIRED COMPONENTS DBus)
 endif (UNIX)
-
-qt_standard_project_setup()
+set(CMAKE_AUTOUIC ON)
+set(CMAKE_AUTOMOC ON)
 
 qt_add_resources(simple64-gui imageresources
     FILES icons/simple64.svg
@@ -68,7 +68,7 @@ if (UNIX)
     target_sources(simple64-gui PRIVATE
         osal/osal_dynamiclib_unix.c
     )
-    target_link_libraries(simple64-gui PRIVATE Qt6::DBus)
+    target_link_libraries(simple64-gui PRIVATE Qt::DBus)
 endif (UNIX)
 
 set_property(TARGET simple64-gui PROPERTY C_VISIBILITY_PRESET hidden)
@@ -86,7 +86,7 @@ endif()
 if (NOT APPLE)
     target_link_options(simple64-gui PRIVATE -Wl,-Bsymbolic -Wl,-Bsymbolic-functions)
 endif (NOT APPLE)
-target_link_libraries(simple64-gui PRIVATE Qt6::Widgets Qt6::WebSockets ${CMAKE_DL_LIBS})
+target_link_libraries(simple64-gui PRIVATE Qt::Widgets Qt::WebSockets ${CMAKE_DL_LIBS})
 
 set_target_properties(simple64-gui PROPERTIES
     WIN32_EXECUTABLE ON
