diff --git a/build.sh b/build.sh
index 24d073d..9d14e3c 100755
--- a/build.sh
+++ b/build.sh
@@ -34,17 +34,17 @@ VERBOSE=1 cmake --build .
 cp ./*mupen64plus"${suffix}" "${install_dir}"
 cp "${base_dir}/mupen64plus-core/data/"* "${install_dir}"
 
-mkdir -p "${base_dir}/simple64-input-raphnetraw/build"
-cd "${base_dir}/simple64-input-raphnetraw/build"
-cmake -G Ninja -DCMAKE_BUILD_TYPE="${RELEASE_TYPE}" ..
-VERBOSE=1 cmake --build .
-cp simple64-input-raphnetraw.* "${install_dir}"
-
-mkdir -p "${base_dir}/simple64-input-qt/build"
-cd "${base_dir}/simple64-input-qt/build"
-cmake -G Ninja -DCMAKE_BUILD_TYPE="${RELEASE_TYPE}" ..
-VERBOSE=1 cmake --build .
-cp simple64-input-qt.* "${install_dir}"
+# mkdir -p "${base_dir}/simple64-input-raphnetraw/build"
+# cd "${base_dir}/simple64-input-raphnetraw/build"
+# cmake -G Ninja -DCMAKE_BUILD_TYPE="${RELEASE_TYPE}" ..
+# VERBOSE=1 cmake --build .
+# cp simple64-input-raphnetraw.* "${install_dir}"
+
+# mkdir -p "${base_dir}/simple64-input-qt/build"
+# cd "${base_dir}/simple64-input-qt/build"
+# cmake -G Ninja -DCMAKE_BUILD_TYPE="${RELEASE_TYPE}" ..
+# VERBOSE=1 cmake --build .
+# cp simple64-input-qt.* "${install_dir}"
 
 mkdir -p "${base_dir}/simple64-audio-sdl2/build"
 cd "${base_dir}/simple64-audio-sdl2/build"
@@ -141,11 +141,11 @@ elif [[ ${UNAME} == *"Darwin"* ]]; then
 else
   if [[ "${PLATFORM}" == "aarch64" ]]; then
     my_os=linux_aarch64
-    cp "${base_dir}/simple64-input-qt/vosk/aarch64/libvosk.so" "${install_dir}"
+    # cp "${base_dir}/simple64-input-qt/vosk/aarch64/libvosk.so" "${install_dir}"
   else
     my_os=linux_x86_64
-    cp "${base_dir}/simple64-gui/discord/libdiscord_game_sdk.so" "${install_dir}"
-    cp "${base_dir}/simple64-input-qt/vosk/x86_64/libvosk.so" "${install_dir}"
+    # cp "${base_dir}/simple64-gui/discord/libdiscord_game_sdk.so" "${install_dir}"
+    # cp "${base_dir}/simple64-input-qt/vosk/x86_64/libvosk.so" "${install_dir}"
   fi
 fi
 
