diff -rupN PPSSPPSDL.orig/SDL/SDLMain.cpp PPSSPPSDL/SDL/SDLMain.cpp
--- PPSSPPSDL.orig/SDL/SDLMain.cpp	2022-06-06 07:03:21.558185006 -0400
+++ PPSSPPSDL/SDL/SDLMain.cpp	2022-06-06 07:29:43.562901323 -0400
@@ -781,9 +781,7 @@ int main(int argc, char *argv[]) {
 		printf("Init from thread error: '%s'\n", error_message.c_str());
 	}
 
-#ifdef MOBILE_DEVICE
 	SDL_ShowCursor(SDL_DISABLE);
-#endif
 
 	if (!useEmuThread) {
 		NativeInitGraphics(graphicsContext);
@@ -874,12 +872,7 @@ int main(int argc, char *argv[]) {
 						g_Config.iForceFullScreen = -1;
 					}
 
-					// Hide/Show cursor correctly toggling fullscreen
-					if (lastUIState == UISTATE_INGAME && fullscreen && !g_Config.bShowTouchControls) {
-						SDL_ShowCursor(SDL_DISABLE);
-					} else if (lastUIState != UISTATE_INGAME || !fullscreen) {
-						SDL_ShowCursor(SDL_ENABLE);
-					}
+					SDL_ShowCursor(SDL_DISABLE);
 					break;
 				}
 
@@ -1142,10 +1135,7 @@ int main(int argc, char *argv[]) {
 #if !defined(MOBILE_DEVICE)
 		if (lastUIState != GetUIState()) {
 			lastUIState = GetUIState();
-			if (lastUIState == UISTATE_INGAME && g_Config.UseFullScreen() && !g_Config.bShowTouchControls)
-				SDL_ShowCursor(SDL_DISABLE);
-			if (lastUIState != UISTATE_INGAME || !g_Config.UseFullScreen())
-				SDL_ShowCursor(SDL_ENABLE);
+			SDL_ShowCursor(SDL_DISABLE);
 		}
 #endif
 
