diff --git a/yabause/src/CMakeLists.txt b/yabause/src/CMakeLists.txt
index 6a088226..efa2d8de 100755
--- a/yabause/src/CMakeLists.txt
+++ b/yabause/src/CMakeLists.txt
@@ -388,7 +388,7 @@ if (YAB_WANT_OPENGL AND (YAB_RGB STREQUAL ""))
 
 	if (OPENGL_SUPPORTED)
 		add_definitions(-DHAVE_LIBGL=1)
-		set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} ${OPENGL_CURRENT_LIBRARIES})
+		set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} ${OPENGL_CURRENT_LIBRARIES} -lglfw -lGLU -lGL -lGLEW)
 
 		include_directories(./ ./tools/nanovg)
 		set(kronos_SOURCES ${kronos_SOURCES}
