diff -rupN vice.orig/Makefile vice/Makefile
--- vice.orig/Makefile	2023-09-10 22:23:12.000000000 +0000
+++ vice/Makefile	2023-09-16 12:43:09.816419382 +0000
@@ -416,7 +416,6 @@ COMMONFLAGS += -DCORE_NAME=\"$(EMUTYPE)\
 include Makefile.common
 
 OBJECTS     += $(patsubst %.cpp,%.o,$(SOURCES_CXX:.cc=.o)) $(SOURCES_C:.c=.o)
-PLATFLAGS   := $(CFLAGS)
 CXXFLAGS    += $(fpic) $(INCFLAGS) $(COMMONFLAGS)
 CFLAGS      += $(fpic) $(INCFLAGS) $(COMMONFLAGS)
 LDFLAGS     += -lm $(fpic)
@@ -437,12 +436,6 @@ ifeq ($(platform), theos_ios)
 	${LIBRARY_NAME}_FILES = $(SOURCES_CXX) $(SOURCES_C)
 	include $(THEOS_MAKE_PATH)/library.mk
 else
-
-default:
-	$(info CFLAGS: $(PLATFLAGS) $(COMMONFLAGS))
-	$(info -------)
-	$(MAKE) $(TARGET)
-
 all: $(TARGET)
 
 -include $(OBJECTS:.o=.d))
@@ -490,5 +483,5 @@ objectclean:
 targetclean:
 	rm -f $(TARGET)
 
-.PHONY: all clean objectclean targetclean
+.PHONY: clean
 endif
