diff --git a/src/duckstation-libretro/libretro_host_interface.cpp b/src/duckstation-libretro/libretro_host_interface.cpp
index bb5f647e..35ef7f4a 100644
--- a/src/duckstation-libretro/libretro_host_interface.cpp
+++ b/src/duckstation-libretro/libretro_host_interface.cpp
@@ -175,9 +175,9 @@ std::string LibretroHostInterface::GetShaderCacheBasePath() const
     }
   }

-  // Use a directory named "duckstation_cache" in the save/system directory.
+  // Use a directory named ".duckstation_cache" in the save/system directory.
   std::string shader_cache_path = StringUtil::StdStringFromFormat(
-    "%s" FS_OSPATH_SEPARATOR_STR "duckstation_cache" FS_OSPATH_SEPARATOR_STR, save_directory_ptr);
+    "%s" FS_OSPATH_SEPARATOR_STR ".duckstation_cache" FS_OSPATH_SEPARATOR_STR, save_directory_ptr);
   if (!FileSystem::DirectoryExists(shader_cache_path.c_str()) &&
       !FileSystem::CreateDirectory(shader_cache_path.c_str(), false))
   {
