diff -rupN flycast.orig/Makefile flycast/Makefile
--- flycast.orig/Makefile	2022-06-18 13:29:52.675787687 -0400
+++ flycast/Makefile	2022-06-18 13:53:23.564624927 -0400
@@ -403,26 +403,30 @@ else ifeq ($(platform), arm64_cortex_a53
 
 #######################################
 
-# ARM64 SM1 Odroid C4
-else ifeq ($(platform), odroidc4)
+# ARM64 RK3588
+else ifeq ($(platform), RK3588)
 	EXT ?= so
 	TARGET := $(TARGET_NAME)_libretro.$(EXT)
 	SHARED += -shared -Wl,--version-script=link.T
-	LDFLAGS +=  -Wl,--no-undefined
 	fpic = -fPIC
 	LIBS += -lrt
 	ARM_FLOAT_ABI_HARD = 0
 	FORCE_GLES = 1
 	SINGLE_PREC_FLAGS = 1
-	CPUFLAGS += -DHOST_CPU=0x20000006 -DTARGET_LINUX_ARMv8 -frename-registers
-	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 $(CPUFLAGS)
-	CXXFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 $(CPUFLAGS)
-	ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
+	HAVE_LTCG = 0
+	HAVE_OPENMP = 0
+	CFLAGS += -Ofast \
+		-fuse-linker-plugin \
+		-fno-stack-protector -fno-ident -fomit-frame-pointer \
+		-fmerge-all-constants -ffast-math -funroll-all-loops \
+		-mcpu=cortex-a76 -mtune=cortex-a76
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76 -Ofast -flto -fuse-linker-plugin
 	PLATFORM_EXT := unix
+	CORE_DEFINES += -DLOW_END -DLOW_RES
 	WITH_DYNAREC=arm64
+	PLATFORM_EXT := unix
 	HAVE_GENERIC_JIT = 0
-	HAVE_VULKAN = 0
-	HAVE_LTCG = 0
