diff --git a/src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp b/src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp
index 3a52f64..c6fca57 100644
--- a/src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp
+++ b/src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp
@@ -995,7 +995,7 @@ void LatteRenderTarget_copyToBackbuffer(LatteTextureView* textureView, bool isPa
 	}
 	cemu_assert(shader);
 	g_renderer->DrawBackbufferQuad(textureView, shader, filter==LatteTextureView::MagFilter::kLinear, imageX, imageY, imageWidth, imageHeight, isPadView, clearBackground);
-	g_renderer->HandleScreenshotRequest(textureView, isPadView);
+	//g_renderer->HandleScreenshotRequest(textureView, isPadView);
 	if (!g_renderer->ImguiBegin(!isPadView))
 		return;
 	swkbd_render(!isPadView);
@@ -1015,13 +1015,13 @@ void LatteRenderTarget_itHLECopyColorBufferToScanBuffer(MPTR colorBufferPtr, uin
 		return;
 	}
 
-	const bool tabPressed = gui_isKeyDown(PlatformKeyCodes::TAB);
-	const bool ctrlPressed = gui_isKeyDown(PlatformKeyCodes::LCONTROL);
+	const bool lctrlPressed = gui_isKeyDown(PlatformKeyCodes::LCONTROL);
+	const bool rctrlPressed = gui_isKeyDown(PlatformKeyCodes::RCONTROL);
 
-	bool showDRC = swkbd_hasKeyboardInputHook() == false && tabPressed;
+	bool showDRC = swkbd_hasKeyboardInputHook() == false && (lctrlPressed || rctrlPressed);
 	bool& alwaysDisplayDRC = LatteGPUState.alwaysDisplayDRC;
 
-	if (ctrlPressed && tabPressed)
+	if (lctrlPressed || rctrlPressed)
 	{
 		if (ctrlTabHotkeyPressed == false)
 		{
@@ -1033,7 +1033,7 @@ void LatteRenderTarget_itHLECopyColorBufferToScanBuffer(MPTR colorBufferPtr, uin
 		ctrlTabHotkeyPressed = false;
 
 	if (alwaysDisplayDRC)
-		showDRC = !tabPressed;
+		showDRC = (!lctrlPressed || !rctrlPressed);
 
 	if (!showDRC)
 	{
