diff -rupN wayland-1.21.0.orig/src/meson.build wayland-1.21.0/src/meson.build
--- wayland-1.21.0.orig/src/meson.build	2022-08-27 16:11:57.858458871 -0400
+++ wayland-1.21.0/src/meson.build	2022-08-27 16:20:31.739044226 -0400
@@ -72,16 +72,6 @@ if get_option('scanner')
 		filebase: 'wayland-scanner'
 	)
 
-	if meson.can_run_host_binaries()
-		meson.override_find_program('wayland-scanner', wayland_scanner)
-	endif
-endif
-
-if meson.is_cross_build() or not get_option('scanner')
-	scanner_dep = dependency('wayland-scanner', native: true, version: meson.project_version())
-	wayland_scanner_for_build = find_program(scanner_dep.get_variable(pkgconfig: 'wayland_scanner'))
-else
-	wayland_scanner_for_build = wayland_scanner
 endif
 
 if get_option('libraries')
@@ -137,7 +127,7 @@ if get_option('libraries')
 		target = custom_target(
 			target_name,
 			command: [
-				wayland_scanner_for_build, '-s', scanner_args,
+				'wayland-scanner', '-s', scanner_args,
 				'@INPUT@', '@OUTPUT@'
 			],
 			input: wayland_protocol_xml,
@@ -152,7 +142,7 @@ if get_option('libraries')
 	wayland_protocol_c = custom_target(
 		'protocol source',
 		command: [
-			wayland_scanner_for_build, '-s', 'public-code', '@INPUT@', '@OUTPUT@'
+			'wayland-scanner', '-s', 'public-code', '@INPUT@', '@OUTPUT@'
 		],
 		input: wayland_protocol_xml,
 		output: 'wayland-protocol.c'
