diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce2dd4a..87d8c5e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@ project(parallel-rdp LANGUAGES CXX C)
 set(OpenGL_GL_PREFERENCE GLVND)
 
 set(NAME_PLUGIN_M64P "mupen64plus-video-parallel")
+if(DEFINED APIDIR)
+    include_directories("${APIDIR}")
+endif()
 include_directories(../mupen64plus-core/src/api)
 add_definitions(-DM64P_PLUGIN_API)
 
