diff -rupN linux-6.1.6.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c linux-6.1.6/drivers/gpu/drm/drm_panel_orientation_quirks.c
--- linux-6.1.6.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c	2023-01-19 14:54:26.353935375 -0500
+++ linux-6.1.6/drivers/gpu/drm/drm_panel_orientation_quirks.c	2023-01-22 06:24:30.604966825 -0500
@@ -164,12 +164,24 @@ static const struct dmi_system_id orient
 		  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "T103HAF"),
 		},
 		.driver_data = (void *)&lcd800x1280_rightside_up,
-	}, {	/* AYA NEO 2021 */
+        }, {    /* AYANEO AYANEO 2 */
+                .matches = {
+                  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "AYANEO"),
+                  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "AYANEO 2"),
+                },
+                .driver_data = (void *)&lcd1200x1920_rightside_up,
+	}, {	/* AYA NEO 2021 and variants */
 		.matches = {
-		  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "AYADEVICE"),
-		  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "AYA NEO 2021"),
-		},
-		.driver_data = (void *)&lcd800x1280_rightside_up,
+		  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "AYANEO"),
+		  DMI_MATCH(DMI_PRODUCT_NAME, "2021"),
+                },
+                .driver_data = (void *)&lcd800x1280_rightside_up,
+        }, {    /* AYA NEO Founder */
+                .matches = {
+                  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "AYA NEO"),
+                  DMI_MATCH(DMI_PRODUCT_NAME, "AYA NEO Founder"),
+                },
+                .driver_data = (void *)&lcd800x1280_rightside_up,
 	}, {	/* AYA NEO AIR */
 		.matches = {
 		  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "AYANEO"),
@@ -182,6 +194,12 @@ static const struct dmi_system_id orient
 		  DMI_MATCH(DMI_BOARD_NAME, "NEXT"),
 		},
 		.driver_data = (void *)&lcd800x1280_rightside_up,
+        }, {    /* AYA NEO GEEK */
+                .matches = {
+                  DMI_EXACT_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+                  DMI_MATCH(DMI_BOARD_NAME, "GEEK"),
+                },
+                .driver_data = (void *)&lcd800x1280_rightside_up,
 	}, {	/* Chuwi HiBook (CWI514) */
 		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "Hampoo"),
