diff --git a/glib/gnulib/meson.build b/glib/gnulib/meson.build
index 38b530aa0..4408a3dc6 100644
--- a/glib/gnulib/meson.build
+++ b/glib/gnulib/meson.build
@@ -302,12 +302,12 @@ else
   gl_cv_func_frexpl_broken_beyond_repair = true
 endif
 
-if not gl_cv_func_frexp_works and gl_cv_func_frexp_broken_beyond_repair
-  error ('frexp() is missing or broken beyond repair, and we have nothing to replace it with')
-endif
-if not gl_cv_func_frexpl_works and gl_cv_func_frexpl_broken_beyond_repair
-  error ('frexpl() is missing or broken beyond repair, and we have nothing to replace it with')
-endif
+#if not gl_cv_func_frexp_works and gl_cv_func_frexp_broken_beyond_repair
+#  error ('frexp() is missing or broken beyond repair, and we have nothing to replace it with')
+#endif
+#if not gl_cv_func_frexpl_works and gl_cv_func_frexpl_broken_beyond_repair
+#  error ('frexpl() is missing or broken beyond repair, and we have nothing to replace it with')
+#endif
 
 math_h_config.set ('REPLACE_FREXP', gl_cv_func_frexp_works ? 0 : 1)
 math_h_config.set ('REPLACE_FREXPL', gl_cv_func_frexpl_works ? 0 : 1)
