diff -rupN vitaquake2.orig/Makefile vitaquake2/Makefile
--- vitaquake2.orig/Makefile	2023-09-15 16:57:45.991890996 +0000
+++ vitaquake2/Makefile	2023-09-15 20:08:45.856895708 +0000
@@ -76,6 +76,14 @@ else ifeq ($(platform), linux-portable)
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T
 	LIBM :=
 
+else ifeq ($(platform), @DEVICE@)
+    TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
+    GLES := 1
+    GLES31 := 1
+
 else ifneq (,$(findstring rockchip,$(platform)))
    EXT ?= so
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
