diff --git a/src/osal_preproc.h b/src/osal_preproc.h
index c81525f..e62bf9c 100644
--- a/src/osal_preproc.h
+++ b/src/osal_preproc.h
@@ -57,7 +57,11 @@
 
   #include <limits.h>  // for PATH_MAX
 
-  #define OSAL_DEFAULT_DYNLIB_FILENAME "libmupen64plus.so.2"
+  #ifdef SIMPLECORE
+    #define OSAL_DEFAULT_DYNLIB_FILENAME "libsimple64.so.2"
+  #else
+    #define OSAL_DEFAULT_DYNLIB_FILENAME "libmupen64plus.so.2"
+  #endif
   #define OSAL_DIR_SEPARATOR           '/'
   #define OSAL_CURRENT_DIR             "./"
   #define OSAL_DLL_EXTENSION           ".so"
